---
title: "GSE194289 Analysis"
author: "David Cooper"
date: "2022-10-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(GEOquery)
library(DESeq2)
library(tidyverse)
library(fgsea)
library(data.table)
library(preprocessCore)
library(pheatmap)
library(apeglm)
library(EnhancedVolcano)
library(biomaRt)
source("read_cpdb_tab.R")

```

## Data Acquisition

Use GEOquery to obtain the GEO series GSE194289.

```{r Get ExpressionSets}
GSE194289=getGEO("GSE194289")
GSEname="GSE194289"
getGEOSuppFiles(GSEname)
untar("GSE194289/GSE194289_RAW.tar",exdir = "GSE194289_GSM")
Files=list.files("GSE194289_GSM/",full.names = TRUE)
getwd()
for (i in Files) {
  gunzip(i)
}
CountFiles=list.files("GSE194289_GSM/",full.names = TRUE)
exGSE194289=read.table("GSE194289_GSM/GSM5832474_RRB001-counts.txt",header=FALSE,col.names = c("X","ToRemove"))
for (i in CountFiles){
  exGSE194289=merge(exGSE194289,read.table(i,header=FALSE,col.names = c("X",substr(i,1,10))))
}
exGSE194289=select(exGSE194289,!ToRemove)
rownames(exGSE194289)=exGSE194289$X
fexGSE194289=filter(exGSE194289,str_detect(X,"ENSR"))
fexGSE194289=select(fexGSE194289,!X)
saveRDS(fexGSE194289,"CountTableGSE194289.rds")

```

```{r}
fexGSE194289=readRDS("CountTableGSE194289.rds")
boxplot(log2(fexGSE194289+0.1),las=2)

```


##DESeq2

```{r DESeq2}
GSE194289=getGEO("GSE194289")
SampleData=GSE194289[[1]]@phenoData@data
SampleData$TreatmentRegion=paste(SampleData$`brain region name:ch1`,SampleData$`treatment:ch1`,sep = "_")
SampleData$Treatment=factor(SampleData$`treatment:ch1`,levels = c("Sham","Fluoxetine"))
fexGSE194289=readRDS("CountTableGSE194289.rds")

dds <- DESeqDataSetFromMatrix(countData = fexGSE194289,
                              colData = SampleData,
                              design= ~ Treatment)
dds <- DESeq(dds)
saveRDS(dds,"../FLX_datasets/GSE194289/DESeqDataGroupsGSE194289.rds")
res <- results(dds, name = "Treatment_Fluoxetine_vs_Sham")
saveRDS(res,"../FLX_datasets/GSE194289/DESeqResGSE194289.rds")

dds2 <- DESeqDataSetFromMatrix(countData = fexGSE194289,
                               colData = SampleData,
                               design= ~ TreatmentRegion)
dds2 <- DESeq(dds2)
saveRDS(dds2,"DESeqDataGroupsGSE194289.rds")
resultsNames(dds2) # lists the coefficients

ContrastTable=data.frame(Variable=rep("TreatmentRegion",26),
FlxSamples=levels(as.factor(SampleData$TreatmentRegion))[seq(4,54,2)],
ShamSamples=levels(as.factor(SampleData$TreatmentRegion))[seq(5,55,2)],
Tissue=levels(as.factor(SampleData$`brain region name:ch1`))[3:28])

for (i in 4:nrow(ContrastTable))
  {
  restemp=results(dds2,contrast = c("TreatmentRegion",ContrastTable[i,2],ContrastTable[i,3]))
  saveRDS(restemp,paste("../FLX_datasets/GSE194289/",ContrastTable[i,4],"_DESeqResGSE194289.rds",sep = ""))
}

resArc=results(dds2, contrast = c("TreatmentRegion","Arc_Fluoxetine","Arc_Sham"))
saveRDS(resArc,"../FLX_datasets/GSE194289/Arc_DESeqResGSE194289.rds")

```

```{r volcanoplot, fig.height = 10, fig.width = 12}
#res=readRDS("../FLX_datasets/GSE194289/DESeqResGSE194289.rds")
EnhancedVolcano(res,
  lab = rownames(res),
  x = "log2FoldChange",
  y = "pvalue",
  title = "Sham vs Fluoxetine")

```

## Transcript ID to Gene Symbol

```{r}
ensembl=useEnsembl(biomart="genes",dataset = "rnorvegicus_gene_ensembl")
#attributes = listAttributes(ensembl)
GeneList=getBM(mart = ensembl,attributes = c("ensembl_transcript_id","external_gene_name"))
GenesMouse=getBM(mart=ensembl,attributes = c("ensembl_transcript_id","mmusculus_homolog_ensembl_gene"))
GeneList$MouseEnsembl=GenesMouse$mmusculus_homolog_ensembl_gene[match(GeneList$ensembl_transcript_id,GenesMouse$ensembl_transcript_id)]

```

## Gene Set Enrichment Analysis

Match transcript IDs in dataset with gene names
take transcript with highest expression (counts)
remove blanks
move to gene set enrichment
  Can't use human gene set dataset
  Match with mouse homologs and use mouse dataset (CPDB...mouse)

```{r Remove Duplicate Genes}
res=readRDS("../FLX_datasets/GSE194289/DESeqResGSE194289.rds")
UniqueGenes=as.data.frame(res)%>%
  mutate(GENE_SYMBOL=GeneList$MouseEnsembl[match(rownames(res),GeneList$ensembl_transcript_id)])%>%
  arrange(desc(baseMean))%>%
  filter(!duplicated(GENE_SYMBOL))%>%
  filter(GENE_SYMBOL!=""&!is.na(log2FoldChange)&!is.na(padj))

```  

```{r GSEA using fgsea}
#Pathways list using mouse Ensembl gene IDs
CPDBpaths=read_cpdb_tab("CPDB_pathways_genes_mouse.tab")
#Stats vector treatment vs control (stat and gene symbols) using filtered genes list
pstats=filter(UniqueGenes)$stat
names(pstats)=UniqueGenes$GENE_SYMBOL
#Pathway enrichment
GeneEnrich=fgsea(pathways = CPDBpaths, stats = pstats, minSize  = 10)
head(arrange(GeneEnrich,padj),n=20)
saveRDS(GeneEnrich,"../FLX_datasets/GSE194289/GSEA_GSE194289.rds")
plotEnrichment(CPDBpaths[["Calcium signaling pathway - Mus musculus (mouse) | KEGG"]], pstats)

```

```{r}
CPDBpaths=read_cpdb_tab("CPDB_pathways_genes_mouse.tab")
ResFiles=list.files("../FLX_datasets/GSE194289/")
ResFiles=ResFiles[grepl("_DESeqResGSE194289.rds",ResFiles)]

for (i in 1:length(ResFiles))
  {
  tempres=readRDS(ResFiles[i])
  UniqueGenesTemp=as.data.frame(tempres)%>%
    mutate(GENE_SYMBOL=GeneList$MouseEnsembl[match(rownames(tempres),GeneList$ensembl_transcript_id)])%>%
    arrange(desc(baseMean))%>%
    filter(!duplicated(GENE_SYMBOL))%>%
    filter(GENE_SYMBOL!=""&!is.na(log2FoldChange)&!is.na(padj))
  tempstats=filter(UniqueGenesTemp)$stat
  names(tempstats)=UniqueGenesTemp$GENE_SYMBOL
  GeneEnrichTemp=fgsea(pathways = CPDBpaths, stats = tempstats, minSize  = 10)
  saveRDS(GeneEnrichTemp, paste("../FLX_datasets/GSE194289/",gsub("_DESeqResGSE194289.rds","",ResFiles[i]),"_GSEA_GSE194289.rds",sep = ""))
}

head(arrange(GeneEnrichTemp,padj),n=20)

```



```{r Heatmap GSEA}
TargetGenes=CPDBpaths[["Signaling by GPCR | Reactome"]]
TargetTranscripts=UniqueGenes3$Loci[match(TargetGenes,UniqueGenes3$GENE_SYMBOL)][!is.na(TargetTranscripts)]
ExSubset=filter(fexGSE194289,rownames(fexGSE194289)%in%TargetTranscripts)
pheatmap(ExSubset)#,annotation_col = rColumnNames,scale = "row")


```

