---
title: "GSE202172 Analysis"
author: "Parker Stanley"
date: "05-17-2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(GEOquery)
library(DESeq2)
library(tidyverse)
library(fgsea)
library(data.table)
library(preprocessCore)
library(pheatmap)
library(apeglm)
library(EnhancedVolcano)
library(biomaRt)
source("read_cpdb_tab.R")

```

## Data Acquisition

Use GEOquery to obtain the GEO series. The expression data may not be saved within the GEO expression set data structure, but the feature data (gene labels) and phenotype data (sample labels) should still be present. Use getGEOSuppFiles() to download the supplementary count data files. Filter and relabel that data as necessary to create a table of counts.

```{r Get ExpressionSets}
set.seed(123)
mousedat <- getGEO("GSE202172")[[1]]
mpmatrix <- pData(mousedat)
rownames(mpmatrix) <- c("GH_Veh_INPUT1", "GH_Veh_INPUT2", "SH_Veh_INPUT1", "SH_Veh_INPUT2", "Flx_INPUT1", "Flx_INPUT2", "A_Flx_INPUT1", "A_Flx_INPUT2", "GH_Veh_INPUT3", "GH_Veh_INPUT4", "SH_Veh_INPUT3", "SH_Veh_INPUT4", "Flx_INPUT3", "Flx_INPUT4", "A_Flx_INPUT3", "A_Flx_INPUT4", "GH_Veh_TRAP1", "GH_Veh_TRAP2", "SH_Veh_TRAP1", "SH_Veh_TRAP2", "Flx_TRAP1", "Flx_TRAP2", "A_Flx_TRAP1", "A_Flx_TRAP2", "GH_Veh_TRAP3", "GH_Veh_TRAP4", "SH_Veh_TRAP3", "SH_Veh_TRAP4", "Flx_TRAP3", "Flx_TRAP4", "A_Flx_TRAP3", "A_Flx_TRAP4")
colnames(mpmatrix)[13] <- "Housing" 
colnames(mpmatrix)[14] <- "Treatment"
colnames(mpmatrix)[15] <- "Behavior"
mpmatrix$Housing <- as.factor(mpmatrix$Housing)
mpmatrix$Treatment <- as.factor(mpmatrix$Treatment)
mpmatrix$Behavior <- as.factor(mpmatrix$Behavior)
mpmatrix$Group <- paste(mpmatrix$Housing, mpmatrix$Treatment, mpmatrix$Behavior, sep = "_")
mpmatrix$TreatGroup <- paste(mpmatrix$Housing, mpmatrix$Treatment, sep = "_")

inputpmatrix <- mpmatrix[c("GH_Veh_INPUT1", "GH_Veh_INPUT2", "SH_Veh_INPUT1", "SH_Veh_INPUT2",
                           "Flx_INPUT1", "Flx_INPUT2", "A_Flx_INPUT1", "A_Flx_INPUT2",
                           "GH_Veh_INPUT3", "GH_Veh_INPUT4", "SH_Veh_INPUT3", "SH_Veh_INPUT4",
                           "Flx_INPUT3", "Flx_INPUT4", "A_Flx_INPUT3", "A_Flx_INPUT4"),]
TRAPpmatrix <- mpmatrix[c("GH_Veh_TRAP1", "GH_Veh_TRAP2", "SH_Veh_TRAP1", "SH_Veh_TRAP2",
                          "Flx_TRAP1", "Flx_TRAP2", "A_Flx_TRAP1", "A_Flx_TRAP2", "GH_Veh_TRAP3",
                          "GH_Veh_TRAP4", "SH_Veh_TRAP3", "SH_Veh_TRAP4", "Flx_TRAP3",
                          "Flx_TRAP4", "A_Flx_TRAP3", "A_Flx_TRAP4"),]
#saveRDS(inputpmatrix,"inputpmatrix.rds")
#saveRDS(TRAPpmatrix, "TRAPpmatrix.rds")

#getGEOSuppFiles("GSE202172")
#for (i in list.files("GSE202172/")) {
#  gunzip(i)
#}
#countdata <- list()
#for(file in list.files("GSE202172/")) {
#  countdata[[file]] <- read.table(file)
#}
#countmatrix <- Reduce(function(...) merge(..., all = TRUE, by = "V1"),countdata)
#colnames(countmatrix) <- c("V1", "GH_Veh_INPUT1", "GH_Veh_INPUT2", "SH_Veh_INPUT1", "SH_Veh_INPUT2", "Flx_INPUT1", "Flx_INPUT2", "A_Flx_INPUT1", "A_Flx_INPUT2", "GH_Veh_INPUT3", "GH_Veh_INPUT4", "SH_Veh_INPUT3", "SH_Veh_INPUT4", "Flx_INPUT3", "Flx_INPUT4", "A_Flx_INPUT3", "A_Flx_INPUT4", "GH_Veh_TRAP1", "GH_Veh_TRAP2", "SH_Veh_TRAP1", "SH_Veh_TRAP2", "Flx_TRAP1", "Flx_TRAP2", "A_Flx_TRAP1", "A_Flx_TRAP2", "GH_Veh_TRAP3", "GH_Veh_TRAP4", "SH_Veh_TRAP3", "SH_Veh_TRAP4", "Flx_TRAP3", "Flx_TRAP4", "A_Flx_TRAP3", "A_Flx_TRAP4")
#rownames(countmatrix) <- countmatrix$V1
#countmatrix$V1 <- NULL
#inputcounts <- countmatrix[, c("GH_Veh_INPUT1", "GH_Veh_INPUT2", "SH_Veh_INPUT1", "SH_Veh_INPUT2", "Flx_INPUT1", "Flx_INPUT2", "A_Flx_INPUT1",  "A_Flx_INPUT2", "GH_Veh_INPUT3", "GH_Veh_INPUT4", "SH_Veh_INPUT3", "SH_Veh_INPUT4", "Flx_INPUT3", "Flx_INPUT4", "A_Flx_INPUT3", "A_Flx_INPUT4")]
#TRAPcounts <- countmatrix[, c("GH_Veh_TRAP1", "GH_Veh_TRAP2", "SH_Veh_TRAP1", "SH_Veh_TRAP2", "Flx_TRAP1", "Flx_TRAP2", "A_Flx_TRAP1",  "A_Flx_TRAP2", "GH_Veh_TRAP3", "GH_Veh_TRAP4", "SH_Veh_TRAP3", "SH_Veh_TRAP4", "Flx_TRAP3", "Flx_TRAP4", "A_Flx_TRAP3", "A_Flx_TRAP4")]
#saveRDS(inputcounts,"inputcountsGSE202172.rds")
#saveRDS(TRAPcounts, "TRAPcountsGSE202172.rds")

inputcounts=readRDS("inputcountsGSE202172.rds")
TRAPcounts=readRDS("TRAPcountsGSE202172.rds")

```

## Quality Control Analyses

Compare samples with boxplots or quantile plots to assess the quality across samples. Normalized datasets should have the same median across the samples. Boxplots will show if there are any samples that differ greatly from the rest. Boxplots will often be most useful for log transformed expression data.

```{r Boxplots}
boxplot(log2(1+inputcounts))
boxplot(log2(1+TRAPcounts))

```

## Remove Low-Expressed Genes

To focus on gene loci with expression that can be compared across samples, it may be useful to remove gene loci without expression above a threshold in at least half of the samples.

```{r Filter Low-Expressed Genes}
inputcounts <- inputcounts[rowSums(inputcounts) > 5,]
TRAPcounts <- TRAPcounts[rowSums(TRAPcounts) > 5,]

```

## Differential Expression Analysis

Use DESeq2 to analyze expression.

```{r Differential Expression Analysis DESeq2}
#DESeq2 for Input Count Data
dds_input <- DESeqDataSetFromMatrix(countData = inputcounts, colData = inputpmatrix, design = ~ Group)
levels(dds_input$Group)
levels(dds_input$Group) <- c("GH_Veh_Norm", "SH_Flx_Anx", "SH_Flx_Norm", "SH_Veh_Norm")
dds_input <- DESeq(dds_input)

INPUT.SH_Veh.vs.GH_Veh <- results(dds_input, contrast = c("Group", "SH_Veh_Norm", "GH_Veh_Norm"))
INPUT.Flx_Sh.vs.SH_Veh <- results(dds_input, contrast = c("Group", "SH_Flx_Norm", "SH_Veh_Norm"))
INPUT.Flx_Anx.vs.SH_Veh <- results(dds_input, contrast = c("Group", "SH_Flx_Anx", "SH_Veh_Norm"))
INPUT.Flx_Norm.vs.Flx_Anx <- results(dds_input, contrast = c("Group", "SH_Flx_Norm", "SH_Flx_Anx"))

dds_input2 <- DESeqDataSetFromMatrix(countData = inputcounts, colData = inputpmatrix, design = ~ TreatGroup)
levels(dds_input2$TreatGroup)
levels(dds_input2$TreatGroup) <- c("GH_Veh", "SH_Flx", "SH_Veh")
dds_input2 <- DESeq(dds_input2)

INPUT.Flx.vs.Veh <- results(dds_input2, contrast = c("TreatGroup", "SH_Flx", "SH_Veh"))

saveRDS(dds_input,"../FLX_datasets/GSE202172/dds_input_GSE202172.rds")
saveRDS(dds_input2,"../FLX_datasets/GSE202172/dds_input2_GSE202172.rds")
saveRDS(INPUT.SH_Veh.vs.GH_Veh, "../FLX_datasets/GSE202172/DESeqINPUT.SH_Veh.vs.GH_Veh_GSE202172.rds")
saveRDS(INPUT.Flx_Sh.vs.SH_Veh, "../FLX_datasets/GSE202172/DESeqINPUT.Flx_Sh.vs.SH_Veh_GSE202172.rds")
saveRDS(INPUT.Flx_Anx.vs.SH_Veh, "../FLX_datasets/GSE202172/DESeqINPUT.Flx_Anx.vs.SH_Veh_GSE202172.rds")
saveRDS(INPUT.Flx_Norm.vs.Flx_Anx, "../FLX_datasets/GSE202172/DESeqINPUT.Flx_Norm.vs.Flx_Anx_GSE202172.rds")
saveRDS(INPUT.Flx.vs.Veh, "../FLX_datasets/GSE202172/DESeqINPUT.Flx.vs.Veh_GSE202172.rds")

#DESeq2 for TRAP Count Data
dds_TRAP <- DESeqDataSetFromMatrix(countData = TRAPcounts, colData = TRAPpmatrix, design = ~ Group)
levels(dds_TRAP$Group)
levels(dds_TRAP$Group) <- c("GH_Veh_Norm", "SH_Flx_Anx", "SH_Flx_Norm", "SH_Veh_Norm")
dds_TRAP <- DESeq(dds_TRAP)

TRAP.SH_Veh.vs.GH_Veh <- results(dds_TRAP, contrast = c("Group", "SH_Veh_Norm", "GH_Veh_Norm"))
TRAP.Flx_Sh.vs.SH_Veh <- results(dds_TRAP, contrast = c("Group", "SH_Flx_Norm", "SH_Veh_Norm"))
TRAP.Flx_Anx.vs.SH_Veh <- results(dds_TRAP, contrast = c("Group", "SH_Flx_Anx", "SH_Veh_Norm"))
TRAP.Flx_Norm.vs.Flx_Anx <- results(dds_TRAP, contrast = c("Group", "SH_Flx_Norm", "SH_Flx_Anx"))

dds_TRAP2 <- DESeqDataSetFromMatrix(countData = TRAPcounts, colData = TRAPpmatrix, design = ~ TreatGroup)
levels(dds_TRAP2$TreatGroup)
levels(dds_TRAP2$TreatGroup) <- c("GH_Veh", "SH_Flx", "SH_Veh")
dds_TRAP2 <- DESeq(dds_TRAP2)

TRAP.Flx.vs.Veh <- results(dds_TRAP2, contrast = c("TreatGroup", "SH_Flx", "SH_Veh"))

saveRDS(dds_TRAP, "../FLX_datasets/GSE202172/dds_TRAP_GSE202172.rds")
saveRDS(dds_TRAP2,"../FLX_datasets/GSE202172/dds_TRAP2_GSE202172.rds")
saveRDS(TRAP.SH_Veh.vs.GH_Veh, "../FLX_datasets/GSE202172/DESeqTRAP.SH_Veh.vs.GH_Veh_GSE202172.rds")
saveRDS(TRAP.Flx_Sh.vs.SH_Veh, "../FLX_datasets/GSE202172/DESeqTRAP.Flx_Sh.vs.SH_Veh_GSE202172.rds")
saveRDS(TRAP.Flx_Anx.vs.SH_Veh, "../FLX_datasets/GSE202172/DESeqTRAP.Flx_Anx.vs.SH_Veh_GSE202172.rds")
saveRDS(TRAP.Flx_Norm.vs.Flx_Anx, "../FLX_datasets/GSE202172/DESeqTRAP.Flx_Norm.vs.Flx_Anx_GSE202172.rds")
saveRDS(TRAP.Flx.vs.Veh, "../FLX_datasets/GSE202172/DESeqTRAP.Flx.vs.Veh_GSE202172.rds")

```

## Differential Expression Plots

Principal component analysis (PCA). DESeq results need to be transformed to use in a PCA plot: vsd=varianceStabilizingTransformation(dds) plotPCA(outside=TRUE) will place the legend on the outside of the plot.

```{r PCA}
input.vsd <- vst(dds_input, blind = FALSE)
TRAP.vsd <- vst(dds_TRAP, blind = FALSE)

plotPCA(input.vsd, intgroup = "Group")
plotPCA(TRAP.vsd, intgroup = "Group")

```

Volcano plots or MA plots can be used to visualize differential expression results.

```{r Volcano Plots}
EnhancedVolcano(INPUT.SH_Veh.vs.GH_Veh, lab = rownames(INPUT.SH_Veh.vs.GH_Veh), x = 'log2FoldChange', y = 'pvalue', 
                title = "INPUT.SH_Veh.vs.GH_Veh")
EnhancedVolcano(INPUT.Flx_Norm.vs.Flx_Anx, lab = rownames(INPUT.Flx_Norm.vs.Flx_Anx), x = 'log2FoldChange', y = 'pvalue',
                title = "INPUT.Flx_Norm.vs.Flx_Anx")
EnhancedVolcano(INPUT.Flx_Sh.vs.SH_Veh, lab = rownames(INPUT.Flx_Sh.vs.SH_Veh), x = 'log2FoldChange', y = 'pvalue',
                title = "INPUT.Flx_Sh.vs.SH_Veh")

EnhancedVolcano(TRAP.SH_Veh.vs.GH_Veh, lab = rownames(TRAP.SH_Veh.vs.GH_Veh), x = 'log2FoldChange', y = 'pvalue',
                title = "TRAP.SH_Veh.vs.GH_Veh")
EnhancedVolcano(TRAP.Flx_Norm.vs.Flx_Anx, lab = rownames(TRAP.Flx_Norm.vs.Flx_Anx), x = 'log2FoldChange', y = 'pvalue',
                title = "TRAP.Flx_Norm.vs.Flx_Anx")
EnhancedVolcano(TRAP.Flx_Sh.vs.SH_Veh, lab = rownames(TRAP.Flx_Sh.vs.SH_Veh), x = 'log2FoldChange', y = 'pvalue', 
                title = "TRAP.Flx_Sh.vs.SH_Veh")

```

## Processing Differential Expression Data

In order to perform gene set enrichment analysis, the data first needs to be filtered to remove duplicate genes (multiple rows of data that correspond to the same gene) and to ensure that each gene is labeled with an appropriate gene ID. If there are multiple rows of data that correspond to the same gene, we want to keep the row with the highest average expression for that gene across samples.

```{r Remove Duplicate Genes}
#No duplicated Genes

```

Convert transcript IDs to gene symbols using esembl biomarts. See biomaRt_script.R for details on using emsembl biomarts.

```{r Convert Transcript IDs to Gene Symbols}
ensembl <- useEnsembl(biomart="genes",dataset = "mmusculus_gene_ensembl")
MouseGeneList <- getBM(mart=ensembl,attributes = c("ensembl_gene_id", "external_gene_name"))

INPUT.SH_Veh.vs.GH_Veh.df <- as.data.frame(INPUT.SH_Veh.vs.GH_Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
INPUT.SH_Veh.vs.GH_Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(INPUT.SH_Veh.vs.GH_Veh.df), MouseGeneList$external_gene_name)]
INPUT.SH_Veh.vs.GH_Veh.df=filter(INPUT.SH_Veh.vs.GH_Veh.df,!is.na(GeneID))

INPUT.Flx_Sh.vs.SH_Veh.df <- as.data.frame(INPUT.Flx_Sh.vs.SH_Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
INPUT.Flx_Sh.vs.SH_Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(INPUT.Flx_Sh.vs.SH_Veh.df), MouseGeneList$external_gene_name)]
INPUT.Flx_Sh.vs.SH_Veh.df=filter(INPUT.Flx_Sh.vs.SH_Veh.df,!is.na(GeneID))

INPUT.Flx_Anx.vs.SH_Veh.df <- as.data.frame(INPUT.Flx_Anx.vs.SH_Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
INPUT.Flx_Anx.vs.SH_Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(INPUT.Flx_Anx.vs.SH_Veh.df), MouseGeneList$external_gene_name)]
INPUT.Flx_Anx.vs.SH_Veh.df=filter(INPUT.Flx_Anx.vs.SH_Veh.df,!is.na(GeneID))

INPUT.Flx_Norm.vs.Flx_Anx.df <- as.data.frame(INPUT.Flx_Norm.vs.Flx_Anx) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
INPUT.Flx_Norm.vs.Flx_Anx.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(INPUT.Flx_Norm.vs.Flx_Anx.df), MouseGeneList$external_gene_name)]
INPUT.Flx_Norm.vs.Flx_Anx.df=filter(INPUT.Flx_Norm.vs.Flx_Anx.df,!is.na(GeneID))

INPUT.Flx.vs.Veh.df <- as.data.frame(INPUT.Flx.vs.Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
INPUT.Flx.vs.Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(INPUT.Flx.vs.Veh.df), MouseGeneList$external_gene_name)]
INPUT.Flx.vs.Veh.df=filter(INPUT.Flx.vs.Veh.df,!is.na(GeneID))

TRAP.SH_Veh.vs.GH_Veh.df <- as.data.frame(TRAP.SH_Veh.vs.GH_Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
TRAP.SH_Veh.vs.GH_Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(TRAP.SH_Veh.vs.GH_Veh.df), MouseGeneList$external_gene_name)]
TRAP.SH_Veh.vs.GH_Veh.df=filter(TRAP.SH_Veh.vs.GH_Veh.df,!is.na(GeneID))

TRAP.Flx_Sh.vs.SH_Veh.df <- as.data.frame(TRAP.Flx_Sh.vs.SH_Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
TRAP.Flx_Sh.vs.SH_Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(TRAP.Flx_Sh.vs.SH_Veh.df), MouseGeneList$external_gene_name)]
TRAP.Flx_Sh.vs.SH_Veh.df=filter(TRAP.Flx_Sh.vs.SH_Veh.df,!is.na(GeneID))

TRAP.Flx_Anx.vs.SH_Veh.df <- as.data.frame(TRAP.Flx_Anx.vs.SH_Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
TRAP.Flx_Anx.vs.SH_Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(TRAP.Flx_Anx.vs.SH_Veh.df), MouseGeneList$external_gene_name)]
TRAP.Flx_Anx.vs.SH_Veh.df=filter(TRAP.Flx_Anx.vs.SH_Veh.df,!is.na(GeneID))

TRAP.Flx_Norm.vs.Flx_Anx.df <- as.data.frame(TRAP.Flx_Norm.vs.Flx_Anx) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
TRAP.Flx_Norm.vs.Flx_Anx.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(TRAP.Flx_Norm.vs.Flx_Anx.df), MouseGeneList$external_gene_name)]
TRAP.Flx_Norm.vs.Flx_Anx.df=filter(TRAP.Flx_Norm.vs.Flx_Anx.df,!is.na(GeneID))

TRAP.Flx.vs.Veh.df <- as.data.frame(TRAP.Flx.vs.Veh) %>% 
  arrange(desc(baseMean)) %>%
  filter(!is.na(padj))
TRAP.Flx.vs.Veh.df$GeneID <- MouseGeneList$ensembl_gene_id[match(rownames(TRAP.Flx.vs.Veh.df), MouseGeneList$external_gene_name)]
TRAP.Flx.vs.Veh.df=filter(TRAP.Flx.vs.Veh.df,!is.na(GeneID))

```

## Gene Set Enrichment Analysis (GSEA)

Use fgsea to analyze gene set enrichment.

```{r GSEA Analsis}
pathwayDB <- read_cpdb_tab("CPDB_pathways_genes_mouse.tab")

IN.DElist=list()
IN.DEbackground=rownames(as.data.frame(INPUT.SH_Veh.vs.GH_Veh))
TRP.DElist=list()
TRP.DEbackground=rownames(as.data.frame(TRAP.SH_Veh.vs.GH_Veh))

IN.Sh.Gh.Stat <- c(INPUT.SH_Veh.vs.GH_Veh.df$stat)
IN.Sh.Gh.Stat <- set_names(IN.Sh.Gh.Stat, nm = INPUT.SH_Veh.vs.GH_Veh.df$GeneID)
IN.Sh.Gh.fgseaRES <- fgsea(pathways = pathwayDB,
                           stats = IN.Sh.Gh.Stat,
                           minSize = 10, nPermSimple=10000)
IN.DElist$IN.Sh.Gh=rownames(INPUT.SH_Veh.vs.GH_Veh.df[INPUT.SH_Veh.vs.GH_Veh.df$pvalue<0.05,])

IN.Norm.Sh.Stat <- c(INPUT.Flx_Sh.vs.SH_Veh.df$stat)
IN.Norm.Sh.Stat <- set_names(IN.Norm.Sh.Stat, nm = INPUT.Flx_Sh.vs.SH_Veh.df$GeneID)
IN.Norm.Sh.fgseaRES <- fgsea(pathways = pathwayDB,
                             stats = IN.Norm.Sh.Stat,
                             minSize = 10, nPermSimple=10000)
IN.DElist$IN.Norm.Sh=rownames(INPUT.Flx_Sh.vs.SH_Veh.df[INPUT.Flx_Sh.vs.SH_Veh.df$pvalue<0.05,])

IN.Anx.Sh.Stat <- c(INPUT.Flx_Anx.vs.SH_Veh.df$stat)
IN.Anx.Sh.Stat <- set_names(IN.Anx.Sh.Stat, nm = INPUT.Flx_Anx.vs.SH_Veh.df$GeneID)
IN.Anx.Sh.fgseaRES <- fgsea(pathways = pathwayDB,
                            stats = IN.Anx.Sh.Stat,
                            minSize = 10, nPermSimple=10000)
IN.DElist$IN.Anx.Sh=rownames(INPUT.Flx_Anx.vs.SH_Veh.df[INPUT.Flx_Anx.vs.SH_Veh.df$pvalue<0.05,])

IN.Norm.Anx.Stat <- c(INPUT.Flx_Norm.vs.Flx_Anx.df$stat)
IN.Norm.Anx.Stat <- set_names(IN.Norm.Anx.Stat, nm = INPUT.Flx_Norm.vs.Flx_Anx.df$GeneID)
IN.Norm.Anx.fgseaRES <- fgsea(pathways = pathwayDB,
                              stats = IN.Norm.Anx.Stat,
                              minSize = 10, nPermSimple=10000)
IN.DElist$IN.Norm.Anx=rownames(INPUT.Flx_Norm.vs.Flx_Anx.df[INPUT.Flx_Norm.vs.Flx_Anx.df$pvalue<0.05,])

IN.Flx.Sh.Stat <- c(INPUT.Flx.vs.Veh.df$stat)
IN.Flx.Sh.Stat <- set_names(IN.Flx.Sh.Stat, nm = INPUT.Flx.vs.Veh.df$GeneID)
IN.Flx.Sh.fgseaRES <- fgsea(pathways = pathwayDB,
                            stats = IN.Flx.Sh.Stat,
                            minSize = 10, nPermSimple=10000)
IN.DElist$IN.Flx.Sh=rownames(INPUT.Flx.vs.Veh.df[INPUT.Flx.vs.Veh.df$pvalue<0.05,])

#TRAP samples
TRP.Sh.Gh.Stat <- c(TRAP.SH_Veh.vs.GH_Veh.df$stat)
TRP.Sh.Gh.Stat <- set_names(TRP.Sh.Gh.Stat, nm = TRAP.SH_Veh.vs.GH_Veh.df$GeneID)
TRP.Sh.Gh.fgseaRES <- fgsea(pathways = pathwayDB,
                            stats = TRP.Sh.Gh.Stat,
                            minSize = 10, nPermSimple=10000)
TRP.DElist$TRP.Sh.Gh=rownames(TRAP.SH_Veh.vs.GH_Veh.df[TRAP.SH_Veh.vs.GH_Veh.df$pvalue<0.05,])

TRP.Norm.Sh.Stat <- c(TRAP.Flx_Sh.vs.SH_Veh.df$stat)
TRP.Norm.Sh.Stat <- set_names(TRP.Norm.Sh.Stat, nm = TRAP.Flx_Sh.vs.SH_Veh.df$GeneID)
TRP.Norm.Sh.fgseaRES <- fgsea(pathways = pathwayDB,
                              stats = TRP.Norm.Sh.Stat,
                              minSize = 10, nPermSimple=10000)
TRP.DElist$TRP.Norm.Sh=rownames(TRAP.Flx_Sh.vs.SH_Veh.df[TRAP.Flx_Sh.vs.SH_Veh.df$pvalue<0.05,])

TRP.Anx.Sh.Stat <- c(TRAP.Flx_Anx.vs.SH_Veh.df$stat)
TRP.Anx.Sh.Stat <- set_names(TRP.Anx.Sh.Stat, nm = TRAP.Flx_Anx.vs.SH_Veh.df$GeneID)
TRP.Anx.Sh.fgseaRES <- fgsea(pathways = pathwayDB,
                             stats = TRP.Anx.Sh.Stat,
                             minSize = 10, nPermSimple=10000)
TRP.DElist$TRP.Anx.Sh=rownames(TRAP.Flx_Anx.vs.SH_Veh.df[TRAP.Flx_Anx.vs.SH_Veh.df$pvalue<0.05,])

TRP.Norm.Anx.Stat <- c(TRAP.Flx_Norm.vs.Flx_Anx.df$stat)
TRP.Norm.Anx.Stat <- set_names(TRP.Norm.Anx.Stat, nm = TRAP.Flx_Norm.vs.Flx_Anx.df$GeneID)
TRP.Norm.Anx.fgseaRES <- fgsea(pathways = pathwayDB,
                               stats = TRP.Norm.Anx.Stat,
                               minSize = 10, nPermSimple=10000)
TRP.DElist$TRP.Norm.Anx=rownames(TRAP.Flx_Norm.vs.Flx_Anx.df[TRAP.Flx_Norm.vs.Flx_Anx.df$pvalue<0.05,])

TRP.Flx.Sh.Stat <- c(TRAP.Flx.vs.Veh.df$stat)
TRP.Flx.Sh.Stat <- set_names(TRP.Flx.Sh.Stat, nm = TRAP.Flx.vs.Veh.df$GeneID)
TRP.Flx.Sh.fgseaRES <- fgsea(pathways = pathwayDB,
                             stats = TRP.Flx.Sh.Stat,
                             minSize = 10, nPermSimple=10000)
TRP.DElist$TRP.Flx.Sh=rownames(TRAP.Flx.vs.Veh.df[TRAP.Flx.vs.Veh.df$pvalue<0.05,])

#Save GSEA results as an .rds file
saveRDS(IN.Sh.Gh.fgseaRES,"../FLX_datasets/GSE202172/IN.Sh.Gh.fgseaRES_GSE202172.rds")
saveRDS(IN.Norm.Sh.fgseaRES, "../FLX_datasets/GSE202172/IN.Norm.Sh.fgseaRES_GSE202172.rds")
saveRDS(IN.Anx.Sh.fgseaRES, "../FLX_datasets/GSE202172/IN.Anx.Sh.fgseaRES_GSE202172.rds")
saveRDS(IN.Norm.Anx.fgseaRES, "../FLX_datasets/GSE202172/IN.Norm.Anx.fgseaRES_GSE202172.rds")
saveRDS(IN.Flx.Sh.fgseaRES, "../FLX_datasets/GSE202172/IN.Flx.Sh.fgseaRES_GSE202172.rds")

saveRDS(TRP.Sh.Gh.fgseaRES,"../FLX_datasets/GSE202172/TRP.Sh.Gh.fgseaRES_GSE202172.rds")
saveRDS(TRP.Norm.Sh.fgseaRES, "../FLX_datasets/GSE202172/TRP.Norm.Sh.fgseaRES_GSE202172.rds")
saveRDS(TRP.Anx.Sh.fgseaRES, "../FLX_datasets/GSE202172/TRP.Anx.Sh.fgseaRES_GSE202172.rds")
saveRDS(TRP.Norm.Anx.fgseaRES, "../FLX_datasets/GSE202172/TRP.Norm.Anx.fgseaRES_GSE202172.rds")
saveRDS(TRP.Flx.Sh.fgseaRES, "../FLX_datasets/GSE202172/TRP.Flx.Sh.fgseaRES_GSE202172.rds")

write.table(t(plyr::ldply(IN.DElist,rbind)),"../FLX_datasets/GSE202172/GSE202172-IN.DElist.txt",sep = "\t",na = "",quote = FALSE,col.names = FALSE,row.names = FALSE)
write.table(IN.DEbackground,"../FLX_datasets/GSE202172/GSE202172-IN.DEbackground.txt",quote = FALSE,col.names = FALSE,row.names = FALSE)
write.table(t(plyr::ldply(TRP.DElist,rbind)),"../FLX_datasets/GSE202172/GSE202172-TRP.DElist.txt",sep = "\t",na = "",quote = FALSE,col.names = FALSE,row.names = FALSE)
write.table(TRP.DEbackground,"../FLX_datasets/GSE202172/GSE202172-TRP.DEbackground.txt",quote = FALSE,col.names = FALSE,row.names = FALSE)

```

## GSEA plots

An enrichment plot shows the enrichment of genes, from a specific pathway, differentially expressed in one group vs another group.

```{r GSEA Enrichment Plot}
plotEnrichment(pathwayDB[["Ribosome - Mus musculus (mouse) | KEGG"]], 
               IN.Sh.Gh.Stat) + labs(title = "Ribosome - Mus musculus (mouse) | KEGG")

plotEnrichment(pathwayDB[["glycolysis I | MouseCyc"]], 
               IN.Flx.Sh.Stat) + labs(title = "glycolysis I | MouseCyc")

plotEnrichment(pathwayDB[["Spliceosome - Mus musculus (mouse) | KEGG"]], 
               IN.Anx.Sh.Stat) + labs(title = "Spliceosome - Mus musculus (mouse) | KEGG")

plotEnrichment(pathwayDB[["Translation | Reactome"]], TRP.Sh.Gh.Stat) + labs(title = "Translation | Reactome")

plotEnrichment(pathwayDB[["The citric acid (TCA) cycle and respiratory electron transport | Reactome"]],TRP.Flx.Sh.Stat)+labs(title = "The citric acid (TCA) cycle and respiratory electron transport | Reactome")

plotEnrichment(pathwayDB[["Neuronal System | Reactome"]], 
               TRP.Anx.Sh.Stat) + labs(title = "Neuronal System | Reactome")

```

Heatmaps can show patterns of expression for groups of genes across samples. Specific pathways selected after doing GSEA can be analyzed further by creating heatmaps of genes in those pathways.

```{r GSEA Heatmaps}
input.names <- data.frame(Group = colData(dds_input)[,"Group"], row.names = colnames(dds_input))
TRAP.names <- data.frame(Group = colData(dds_TRAP)[,"Group"], row.names = colnames(dds_TRAP))

IN.Gh.Sh.select <- INPUT.SH_Veh.vs.GH_Veh.df$GeneID %in% pathwayDB[["Ribosome - Mus musculus (mouse) | KEGG"]]
pheatmap(assay(input.vsd)[IN.Gh.Sh.select, ], cluster_rows = TRUE, show_rownames = FALSE,
         cluster_cols = TRUE, scale = "row", annotation_col = input.names)

IN.Flx.Veh.select <- INPUT.Flx_Sh.vs.SH_Veh.df$GeneID %in% pathwayDB[["glycolysis I | MouseCyc"]]
pheatmap(assay(input.vsd)[IN.Flx.Veh.select, ], cluster_rows = TRUE, show_rownames = FALSE,
         cluster_cols = TRUE, scale = "row", annotation_col = input.names)

IN.Flx.Anx.select <- INPUT.Flx_Norm.vs.Flx_Anx.df$GeneID %in% pathwayDB[["Spliceosome - Mus musculus (mouse) | KEGG"]]
pheatmap(assay(input.vsd)[IN.Flx.Anx.select, ], cluster_rows = TRUE, show_rownames = FALSE,
         cluster_cols = TRUE, scale = "row", annotation_col = input.names)

TRP.Gh.Sh.select <- TRAP.SH_Veh.vs.GH_Veh.df$GeneID %in% pathwayDB[["Translation | Reactome"]]
pheatmap(assay(TRAP.vsd)[TRP.Gh.Sh.select, ], cluster_rows = TRUE, show_rownames = FALSE,
         cluster_cols = TRUE, scale = "row", annotation_col = TRAP.names)

TRP.Flx.Veh.select <- TRAP.Flx_Sh.vs.SH_Veh.df$GeneID %in% pathwayDB[["The citric acid (TCA) cycle and respiratory electron transport | Reactome"]]
pheatmap(assay(TRAP.vsd)[TRP.Flx.Veh.select, ], cluster_rows = TRUE, show_rownames = FALSE,
         cluster_cols = TRUE, scale = "row", annotation_col = TRAP.names)

TRP.Flx.Anx.select <- TRAP.Flx_Norm.vs.Flx_Anx.df$GeneID %in% pathwayDB[["Neuronal System | Reactome"]]
pheatmap(assay(TRAP.vsd)[TRP.Flx.Anx.select, ], cluster_rows = TRUE, show_rownames = FALSE,
         cluster_cols = TRUE, scale = "row", annotation_col = TRAP.names)

```
