---
title: "Fluoxetine Analysis"
author: "Victoria Gaertig"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(GEOquery)
library(limma)
library(dplyr)
library(fgsea)
library(data.table)
library(ggplot2)
library(pheatmap)
library(biomaRt)
source("read_cpdb_tab.R")

```

## Data Summary

https://www.nature.com/articles/1301497  (mainly abstract and intro)

This study uses phenotypic and genotypic variability of mouse strains to show that there is a genetic component to the behavior and neuronal effects of chronic fluoxeine treatment. The results of this data may support the theory that there is an inherent genetic predisposition that may increase the effectiveness of antidepressants.

```{r get data}
set.seed(123)
dat <- getGEO("GSE56028")[[1]]
#Filter out non-fluoxetine treatments
dat.filtered <- dat[ ,dat$`agent:ch1`=="untreated" | dat$`agent:ch1`=="fluoxetine"]

```

```{r boxplots}
boxplot(exprs(dat))
boxplot(exprs(dat.filtered))

```

## DE Analysis

https://rpubs.com/jrgonzalezISGlobal/transcriptomic_analyses
To continue with the analysis, I had to set up a new design matrix
In this, we are contrasting untreated control versus untreaed stressed and untreated stressed with fluoxetine stressed

```{r differential expression}
matri <- model.matrix(~0+paste(dat.filtered$`agent:ch1`,dat.filtered$`stress:ch1`))
colnames(matri) <- c("fluoxetine_ucms","untreated_control","untreated_ucms")
fit <- lmFit(dat.filtered, matri)
contrast.matrix <- makeContrasts(untreated_ucms-untreated_control,fluoxetine_ucms-untreated_ucms, levels=matri)

fit2 <- contrasts.fit(fit, contrast.matrix)
fit2 <- eBayes(fit2)

saveRDS(fit2,file = "../FLX_datasets/GSE56028/DE_GSE56028.rds")

```

## Results
# comparing control and stressed untreated
Here I am comparing the controlled data and the untreated UCMS data. I then created a histogram to view the results. There is not really any strong differences as the adjusted P.Values come out high

```{r control vs stressed untreated}
compare1 <- topTable(fit2, coef=1, adjust="BH",number=Inf)
hist(compare1$P.Value)
hist(compare1$adj.P.Val)

```

# comparing untreated stressed vs treated stress
Here I am comparing the untreated UCMS data and the treated UCMS data. I then created a histogram to view the results. There is not really any strong differences as the adjusted P.Values come out high

```{r stressed untreated vs stressed treated}
compare2 <- topTable(fit2, coef=2, adjust="BH",number=Inf)
hist(compare2$P.Value)
hist(compare2$adj.P.Val)

```

This provides us with the results from "fit2" to identify which genes are significantly differentially expressed for each contrast

```{r results}
decideTests(fit2)

```

Here I am combining the two comparisons into one big comparison of Control vs Untreated UCMS vs Treated UCMS. I then ploted these results to be able to see how each comparisons and their differences

```{r merged compare}
cbind(compare1,compare2)
merge <- merge(compare1, compare2, by="ID")
plot(merge$logFC.x, merge$logFC.y)

annotation_col <- data.frame(group=paste(dat.filtered$`agent:ch1`,dat.filtered$`stress:ch1`))
rownames(annotation_col) <- colnames(dat.filtered)
pheatmap(dat.filtered[merge$P.Value.x<0.01&merge$P.Value.y<0.01, ], scale="row", annotation_col=annotation_col,)

```

# Pathway Data
I then used pathway data to my two comparisons

```{r pathway data}
compare1$ENS <- gsub(".*ENSRNOT","ENSRNOT",compare1$gene_assignment)
compare1$ENS <- gsub(" .*","",compare1$ENS)

compare2$ENS <- gsub(".*ENSRNOT","ENSRNOT",compare2$gene_assignment)
compare2$ENS <- gsub(" .*","",compare2$ENS)

merge <- merge(compare1, compare2, by="ID")

```

## Changing Names
Here I am changing the names and utilizing the homologous ensembl.

```{r}
#Accessing homologous ensembl gene symbols using biomaRt
ensembl <- useEnsembl(biomart="genes",dataset = "rnorvegicus_gene_ensembl")
#attributes <- listAttributes(ensembl)
MouseHomologGeneList <- getBM(mart=ensembl,attributes = c("ensembl_transcript_id","mmusculus_homolog_ensembl_gene"))

merge$MouseGenes <- MouseHomologGeneList$mmusculus_homolog_ensembl_gene[match(merge$ENS.x, MouseHomologGeneList$ensembl_transcript_id)]

```

## GSEA
Here, I am now ordering the merge data of df and genes by AveExpr of merge in decreasing order.
We then go through and take out all of the duplicates in merged_ordered.
We then take out the t values from the merge_noduplicates and create a new data set called stats.
The names of the stats data set are going to be taken from the column ensembl_gene_id from merged_noduplicates data set.
We are now taking the mouse pathways and making the names match up with each gene data, creating a new data set.
Here I am creating a fgsea pathway for the Control vs Untreated UCMS data data. We are also ordering by p-value.
Here I am creating a fgsea pathway for the Untreated UCMS data vs the Treated UCMS data. We are also ordering by p-value.

```{r fgsea}
mouse_pathways<-read_cpdb_tab("CPDB_pathways_genes_mouse.tab")

DElist=list()

#Control_vs_UntreatedUCMS.x
merge_orderedx <- merge[order(merge$AveExpr.x, decreasing=TRUE),]
merge_noduplicatesx <- filter(merge_orderedx,duplicated(merge_orderedx$MouseGenes)==FALSE & !(merge_orderedx$MouseGenes%in%c("","<NA>")))
statsx <- merge_noduplicatesx$t.x
names(statsx) <- merge_noduplicatesx$MouseGenes
statsx <- statsx[-1]
Control_vs_UntreatedUCMS.x <- fgsea(pathways = mouse_pathways, stats = statsx, minSize  = 10, nPermSimple=10000)
saveRDS(Control_vs_UntreatedUCMS.x,"../FLX_datasets/GSE56028/GSEA_StressVsControl_GSE56028.rds")

#UntreatedUCMS_vs_TreatedUCMS.y
merge_orderedy <- merge[order(merge$AveExpr.y, decreasing=TRUE),]
merge_noduplicatesy <- filter(merge_orderedy,duplicated(merge_orderedy$MouseGenes)==FALSE & !(merge_orderedy$MouseGenes%in%c("","<NA>")))
DElist$SFTU=sapply(strsplit(merge_noduplicatesy[merge_noduplicatesy$gene_assignment.y!="---"&merge_noduplicatesy$P.Value.y<0.05,]$gene_assignment.y,split = " // "), "[[",2)
DEbackground=sapply(strsplit(merge_noduplicatesy[merge_noduplicatesy$gene_assignment.y!="---",]$gene_assignment.y,split = " // "), "[[",2)
statsy <- merge_noduplicatesy$t.y
names(statsy) <- merge_noduplicatesy$MouseGenes
statsy<-statsy[-1]
UntreatedUCMS_vs_TreatedUCMS.y <- fgsea(pathways = mouse_pathways, stats = statsy, minSize  = 10, nPermSimple=10000)
saveRDS(UntreatedUCMS_vs_TreatedUCMS.y,"../FLX_datasets/GSE56028/GSEA_TreatedVsUnreated_GSE56028.rds")

write.table(t(plyr::ldply(DElist,rbind)),"../FLX_datasets/GSE56028/GSE56028-DElist.txt",sep = "\t",na = "",quote = FALSE,col.names = FALSE,row.names = FALSE)
write.table(DEbackground,"../FLX_datasets/GSE56028/GSE56028-DEbackground.txt",quote = FALSE,col.names = FALSE,row.names = FALSE)

```

I then created a heat map of the Control vs Untreated UCMS p-values.

```{r pheatmap Control_vs_UntreatedUCMS.x}
pvalorderedx<-Control_vs_UntreatedUCMS.x[order(Control_vs_UntreatedUCMS.x$pval, decreasing=FALSE),]

annotation_col = data.frame(group=paste(dat.filtered$`agent:ch1`,dat.filtered$`stress:ch1`))
rownames(annotation_col)=colnames(dat.filtered)

pheatmap(dat.filtered[merge$MouseGenes%in%pvalorderedx$leadingEdge[[1]], ], scale="row", annotation_col=annotation_col,)

```

I then created a heat map of the Untreated UCMS vs Treated UCMS p-values.

```{r pheatmap UntreatedUCMS_vs_TreatedUCMS.y}
pvalorderedy<-UntreatedUCMS_vs_TreatedUCMS.y[order(UntreatedUCMS_vs_TreatedUCMS.y$pval, decreasing=FALSE),]

annotation_col = data.frame(group=paste(dat.filtered$`agent:ch1`,dat.filtered$`stress:ch1`))
rownames(annotation_col)=colnames(dat.filtered)

pheatmap(dat.filtered[merge$MouseGenes%in%pvalorderedy$leadingEdge[[1]], ], scale="row", annotation_col=annotation_col,)

```

I also merged the two fgseca data. I then got rid of the leadingedge column, by using the merged function.

```{r}
cbind(Control_vs_UntreatedUCMS.x,UntreatedUCMS_vs_TreatedUCMS.y)

merge_ordered<-merge(Control_vs_UntreatedUCMS.x,UntreatedUCMS_vs_TreatedUCMS.y, by="pathway")
```

The final thing I did was plot "merge_ordered" to view where there are significant similarieties and differences betweeen Control vs Untreate UCMS and Untreated UCMS vs Treated UCMS

```{r}
plot(merge_ordered$NES.x,merge_ordered$NES.y)

```
