---
title: "GSE86392 Analysis"
author: "Victoria Gaertig"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(GEOquery)
library(DESeq2)
library(apeglm)
library(tidyverse)
library(biomaRt)
library(dplyr)
library(fgsea)
library(pheatmap)
source("read_cpdb_tab.R")

```

Gathering the data into the R Markdown

```{r load data}
set.seed(123)
dataset <- getGEO("GSE86392")[[1]]
#getGEOSuppFiles("GSE86392")
#gunzip("GSE86392/GSE86392_expressed_gene_reads.txt.gz")
datanumbers<-read.csv("GSE86392/GSE86392DataSet.csv", row.names = 1)
Sampledata<-data.frame(treatment = (substr(colnames(datanumbers),1,1)), 
                       tissue =(substr(colnames(datanumbers),3,3)),
                       row.names = colnames(datanumbers))
Sampledata$treatment<-factor(x=Sampledata$treatment,levels = c("M","Z","F","A"))

```

Raw data visualization

```{r boxplots}
#Boxplot with all samples
boxplot(log2(datanumbers),las=2)
#Quantile plot with all samples
qs=t(apply(log2(datanumbers),2,quantile,prob=c(0.05,0.25,0.5,0.75,0.95)))
matplot(qs,type="l",lty=1)

```

Differential Expression Analysis

```{r DESeq}
dds<- DESeqDataSetFromMatrix(countData = datanumbers,
                              colData = Sampledata,
                              design= ~ treatment + tissue)
dds<- DESeq(dds)
resultsNames(dds) # lists the coefficients
saveRDS(dds,"../FLX_datasets/GSE86392/DESeqDataGSE86392.rds")

#M (Stressed) vs Z (control)
MvsZ <- results(dds, contrast = c("treatment","M", "Z"))
saveRDS(MvsZ,"../FLX_datasets/GSE86392/Stress_DESeqRes_GSE86392.rds")
plotMA(MvsZ, ylim=c(-2,2))

#M (Stressed) vs A (Acupuncture)
MvsA <- results(dds, contrast = c("treatment","M", "A"))
plotMA(MvsA, ylim=c(-2,2))

#F (Fluoxetine) vs M (Stressed)
MvsF <- results(dds, contrast = c("treatment","F", "M"))
saveRDS(MvsF,"../FLX_datasets/GSE86392/Flx_DESeqRes_GSE86392.rds")
plotMA(MvsA, ylim=c(-2,2))

```

DESeq results Volcano plots

```{r DESeq Volcano plots}
ggplot(data = as.data.frame(MvsZ)) + 
  geom_point(mapping = aes(x = log2FoldChange, y = -log10(pvalue), color = padj<.01)) +
  scale_color_manual(values=c("red","green","blue"))

ggplot(data = as.data.frame(MvsA)) + 
  geom_point(mapping = aes(x = log2FoldChange, y = -log10(pvalue), color = padj<.01)) +
  scale_color_manual(values=c("red","green","blue"))

ggplot(data = as.data.frame(MvsF)) + 
  geom_point(mapping = aes(x = log2FoldChange, y = -log10(pvalue), color = padj<.01)) +
  scale_color_manual(values=c("red","green","blue"))

```

Gene Identifier Conversion

```{r Convert Gene IDs}
ensembl <- useEnsembl(biomart="genes",dataset = "rnorvegicus_gene_ensembl")
MouseHomologGeneList <- getBM(mart=ensembl,attributes = c("ensembl_gene_id","mmusculus_homolog_ensembl_gene"))

DElist=list()

#M vs Z
MvsZ$MouseGenes <- MouseHomologGeneList$mmusculus_homolog_ensembl_gene[match(rownames(MvsZ), MouseHomologGeneList$ensembl_gene_id)]
MvsZ_filtered<-filter(as.data.frame(MvsZ), MouseGenes != "")
MvsZ_baseMean<-arrange(MvsZ_filtered, desc(baseMean))
MvsZ_filtered2<-filter(MvsZ_baseMean, duplicated(MouseGenes)== FALSE&is.na(padj)==FALSE)

#M vs A
MvsA$MouseGenes <- MouseHomologGeneList$mmusculus_homolog_ensembl_gene[match(rownames(MvsA), MouseHomologGeneList$ensembl_gene_id)]
MvsA_filtered<-filter(as.data.frame(MvsA), MouseGenes != "")
MvsA_baseMean<-arrange(MvsA_filtered, desc(baseMean))
MvsA_filtered2<-filter(MvsA_baseMean, duplicated(MouseGenes)== FALSE&is.na(padj)==FALSE)

#F vs M
MvsF$MouseGenes <- MouseHomologGeneList$mmusculus_homolog_ensembl_gene[match(rownames(MvsF), MouseHomologGeneList$ensembl_gene_id)]
MvsF_filtered<-filter(as.data.frame(MvsF), MouseGenes != "")
MvsF_baseMean<-arrange(MvsF_filtered, desc(baseMean))
MvsF_filtered2<-filter(MvsF_baseMean, duplicated(MouseGenes)== FALSE&is.na(padj)==FALSE)

DElist$SFTU=rownames(MvsF_filtered2[MvsF_filtered2$pvalue<0.05,])
DEbackground=rownames(MvsF_filtered2)

write.table(t(plyr::ldply(DElist,rbind)),"../FLX_datasets/GSE86392/GSE86392-DElist.txt",sep = "\t",na = "",quote = FALSE,col.names = FALSE,row.names = FALSE)
write.table(DEbackground,"../FLX_datasets/GSE86392/GSE86392-DEbackground.txt",quote = FALSE,col.names = FALSE,row.names = FALSE)

```

Gene Set Enrichment Analysis

```{r GSEA}
mouse_pathways<-read_cpdb_tab("CPDB_pathways_genes_mouse.tab")

#MvsZ stats
MvsZ_stats<-MvsZ_filtered2$stat
names(MvsZ_stats)<-MvsZ_filtered2$MouseGenes
#MvsZ fgsea
MvsZ_fgsea <- fgsea(pathways = mouse_pathways,stats = MvsZ_stats,minSize  = 10, nPermSimple=10000)
saveRDS(MvsZ_fgsea,"../FLX_datasets/GSE86392/Stress_GSEA_GSE86392.rds")

#MvsF stat
MvsF_stats<-MvsF_filtered2$stat
names(MvsF_stats)<-MvsF_filtered2$MouseGenes
#MvsF fgsea
MvsF_fgsea <- fgsea(pathways = mouse_pathways,stats = MvsF_stats,minSize  = 10, nPermSimple=10000)
saveRDS(MvsF_fgsea,"../FLX_datasets/GSE86392/Flx_GSEA_GSE86392.rds")

#MvsA stats
MvsA_stats<-MvsA_filtered2$stat
names(MvsA_stats)<-MvsA_filtered2$MouseGenes
#MvsA fgsea
MvsA_fgsea <- fgsea(pathways = mouse_pathways,stats = MvsA_stats,minSize  = 10, nPermSimple=10000)

```
