---
title: "MetaDE_Flx"
author: "David Cooper"
date: "2024-1-30"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(vctrs)
library(scales)
library(MetaDE)
library(ggvenn)
library(ggrepel)
library(pheatmap)
library(VennDiagram)
library(biomaRt)
library(grid)
library(gridExtra)

```

## Prep Data Lists for MetaDE

```{r Load Data}
#Comparison Labels
ReplaceLabels=read.csv("ComparisonLabels.csv",header = TRUE)

#Load Gene-level Differential Expression (DE) data
#All comparisons grouped by organism
HumanDEDataList=readRDS("Input RDS/Human_Merged_DE_List.rds")
MouseDEDataList=readRDS("Input RDS/Mouse_Merged_DE_List.rds")
RatDEDataList=readRDS("Input RDS/Rat_Merged_DE_List.rds")

#Organism grouped data to filter into subsets
#Human
HumanDEDataMerge=readRDS("Input RDS/Human_Merged_DE_df.rds")
#Human TvU
HumanDFTUDE=HumanDEDataMerge[,!grepl("GSE83386_Lympho_DFTUXX",colnames(HumanDEDataMerge))&!grepl("GSE83386_Lympho_DFRNXX",colnames(HumanDEDataMerge))]
HumanDFTUDEList=list(p=HumanDFTUDE[,seq(2,ncol(HumanDFTUDE),3)],logFC=HumanDFTUDE[,seq(3,ncol(HumanDFTUDE),3)])
HumanDFTUDEList$Vote=1*(HumanDFTUDEList$p<0.05)*((HumanDFTUDEList$logFC>0)*2-1)

#Mouse
MouseDEDataMerge=readRDS("Input RDS/Mouse_Merged_DE_df.rds")
#Mouse TvU All
MouseFTUDE=MouseDEDataMerge[,grepl("FTU",colnames(MouseDEDataMerge))]
MouseFTUDEList=list(p=MouseFTUDE[,seq(1,ncol(MouseFTUDE),3)],logFC=MouseFTUDE[,seq(2,ncol(MouseFTUDE),3)])
MouseFTUDEList$Vote=1*(MouseFTUDEList$p<0.05)*((MouseFTUDEList$logFC>0)*2-1)
#Mouse TvU No stress
MouseXFTUDE=MouseDEDataMerge[,grepl("XFTU",colnames(MouseDEDataMerge))]
MouseXFTUDEList=list(p=MouseXFTUDE[,seq(1,ncol(MouseXFTUDE),3)],logFC=MouseXFTUDE[,seq(2,ncol(MouseXFTUDE),3)])
MouseXFTUDEList$Vote=1*(MouseXFTUDEList$p<0.05)*((MouseXFTUDEList$logFC>0)*2-1)
#Mouse TvU Stress
MouseSFTUDE=MouseDEDataMerge[,grepl("SFTU",colnames(MouseDEDataMerge))]
MouseSFTUDEList=list(p=MouseSFTUDE[,seq(1,ncol(MouseSFTUDE),3)],logFC=MouseSFTUDE[,seq(2,ncol(MouseSFTUDE),3)])
MouseSFTUDEList$Vote=1*(MouseSFTUDEList$p<0.05)*((MouseSFTUDEList$logFC>0)*2-1)
#Mouse RvN
MouseSFRNDE=MouseDEDataMerge[,grepl("FRN",colnames(MouseDEDataMerge))]
MouseSFRNDEList=list(p=MouseSFRNDE[,seq(1,ncol(MouseSFRNDE),3)],logFC=MouseSFRNDE[,seq(2,ncol(MouseSFRNDE),3)])
MouseSFRNDEList$Vote=1*(MouseSFRNDEList$p<0.05)*((MouseSFRNDEList$logFC>0)*2-1)
#Mouse Brain Only RvN
BrainMouseSFRNDE=MouseDEDataMerge[,grepl("FRN",colnames(MouseDEDataMerge))&!grepl("GSE84184",colnames(MouseDEDataMerge))]
BrainMouseSFRNDEList=list(p=BrainMouseSFRNDE[,seq(1,ncol(BrainMouseSFRNDE),3)],logFC=BrainMouseSFRNDE[,seq(2,ncol(BrainMouseSFRNDE),3)])
BrainMouseSFRNDEList$Vote=1*(BrainMouseSFRNDEList$p<0.05)*((BrainMouseSFRNDEList$logFC>0)*2-1)
#Mouse RvC
MouseSFRCDE=MouseDEDataMerge[,grepl("FRC",colnames(MouseDEDataMerge))]
MouseSFRCDEList=list(p=MouseSFRCDE[,seq(1,ncol(MouseSFRCDE),3)],logFC=MouseSFRCDE[,seq(2,ncol(MouseSFRCDE),3)])
MouseSFRCDEList$Vote=1*(MouseSFRCDEList$p<0.05)*((MouseSFRCDEList$logFC>0)*2-1)
#Mouse NvC
MouseSFNCDE=MouseDEDataMerge[,grepl("FNC",colnames(MouseDEDataMerge))]
MouseSFNCDEList=list(p=MouseSFNCDE[,seq(1,ncol(MouseSFNCDE),3)],logFC=MouseSFNCDE[,seq(2,ncol(MouseSFNCDE),3)])
MouseSFNCDEList$Vote=1*(MouseSFNCDEList$p<0.05)*((MouseSFNCDEList$logFC>0)*2-1)

#Rat
RatDEDataMerge=readRDS("Input RDS/Rat_Merged_DE_df.rds")
#Rat TvU All is unfiltered RatDEDataList
RatDEDataList$Vote=1*(RatDEDataList$p<0.05)*((RatDEDataList$logFC>0)*2-1)
#Rat TvU No stress
RatXFTUDE=RatDEDataMerge[,grepl("XFTU",colnames(RatDEDataMerge))]
RatXFTUDEList=list(p=RatXFTUDE[,seq(1,ncol(RatXFTUDE),3)],logFC=RatXFTUDE[,seq(2,ncol(RatXFTUDE),3)])
RatXFTUDEList$Vote=1*(RatXFTUDEList$p<0.05)*((RatXFTUDEList$logFC>0)*2-1)
#Rat TvU Stress
RatSFTUDE=RatDEDataMerge[,grepl("SFTU",colnames(RatDEDataMerge))]
RatSFTUDEList=list(p=RatSFTUDE[,seq(1,ncol(RatSFTUDE),3)],logFC=RatSFTUDE[,seq(2,ncol(RatSFTUDE),3)])
RatSFTUDEList$Vote=1*(RatSFTUDEList$p<0.05)*((RatSFTUDEList$logFC>0)*2-1)


#Load Pathway-level Gene Set Enrichment Analysis (GSEA) data
#All comparisons grouped by organism
HumanGSEADataList=readRDS("Input RDS/Human_Merged_GSEA_List.rds")
MouseGSEADataList=readRDS("Input RDS/Mouse_Merged_GSEA_List.rds")
RatGSEADataList=readRDS("Input RDS/Rat_Merged_GSEA_List.rds")
AllGSEADataList=readRDS("Input RDS/All_Merged_GSEA_List.rds")

#Organism grouped data to filter into subsets
#Human
HumanGSEADataMerge=readRDS("Input RDS/Human_Merged_GSEA_df.rds")
rownames(HumanGSEADataMerge)=paste(rep("ID",nrow(HumanGSEADataMerge)),1:nrow(HumanGSEADataMerge),sep ="")
HumanGSEADataMergeIDs=dplyr::select(HumanGSEADataMerge,pathway)
#Human TvU
HumanDFTUgsea=HumanGSEADataMerge[,!grepl("GSE83386_Lympho_DFTUXX",colnames(HumanGSEADataMerge))&!grepl("GSE83386_Lympho_DFRNXX",colnames(HumanGSEADataMerge))]
HumanDFTUgseaList=list(p=HumanDFTUgsea[,seq(2,ncol(HumanDFTUgsea),3)],NES=HumanDFTUgsea[,seq(3,ncol(HumanDFTUgsea),3)])
HumanDFTUgseaList$Vote=1*(HumanDFTUgseaList$p<0.05)*((HumanDFTUgseaList$NES>0)*2-1)

#Mouse
MouseGSEADataMerge=readRDS("Input RDS/Mouse_Merged_GSEA_df.rds")
rownames(MouseGSEADataMerge)=paste(rep("ID",nrow(MouseGSEADataMerge)),1:nrow(MouseGSEADataMerge),sep ="")
MouseGSEADataMergeIDs=dplyr::select(MouseGSEADataMerge,pathway)
#Mouse TvU All
MouseFTUgsea=MouseGSEADataMerge[,grepl("FTU",colnames(MouseGSEADataMerge))]
MouseFTUgseaList=list(p=MouseFTUgsea[,seq(1,ncol(MouseFTUgsea),3)],NES=MouseFTUgsea[,seq(2,ncol(MouseFTUgsea),3)])
MouseFTUgseaList$Vote=1*(MouseFTUgseaList$p<0.05)*((MouseFTUgseaList$NES>0)*2-1)
#Mouse TvU No stress
MouseXFTUgsea=MouseGSEADataMerge[,grepl("XFTU",colnames(MouseGSEADataMerge))]
MouseXFTUgseaList=list(p=MouseXFTUgsea[,seq(1,ncol(MouseXFTUgsea),3)],NES=MouseXFTUgsea[,seq(2,ncol(MouseXFTUgsea),3)])
MouseXFTUgseaList$Vote=1*(MouseXFTUgseaList$p<0.05)*((MouseXFTUgseaList$NES>0)*2-1)
#Mouse TvU Stress
MouseSFTUgsea=MouseGSEADataMerge[,grepl("SFTU",colnames(MouseGSEADataMerge))]
MouseSFTUgseaList=list(p=MouseSFTUgsea[,seq(1,ncol(MouseSFTUgsea),3)],NES=MouseSFTUgsea[,seq(2,ncol(MouseSFTUgsea),3)])
MouseSFTUgseaList$Vote=1*(MouseSFTUgseaList$p<0.05)*((MouseSFTUgseaList$NES>0)*2-1)
#Mouse RvN
MouseSFRNgsea=MouseGSEADataMerge[,grepl("FRN",colnames(MouseGSEADataMerge))]
MouseSFRNgseaList=list(p=MouseSFRNgsea[,seq(1,ncol(MouseSFRNgsea),3)],NES=MouseSFRNgsea[,seq(2,ncol(MouseSFRNgsea),3)])
MouseSFRNgseaList$Vote=1*(MouseSFRNgseaList$p<0.05)*((MouseSFRNgseaList$NES>0)*2-1)
#Mouse RvC
MouseSFRCgsea=MouseGSEADataMerge[,grepl("FRC",colnames(MouseGSEADataMerge))]
MouseSFRCgseaList=list(p=MouseSFRCgsea[,seq(1,ncol(MouseSFRCgsea),3)],NES=MouseSFRCgsea[,seq(2,ncol(MouseSFRCgsea),3)])
MouseSFRCgseaList$Vote=1*(MouseSFRCgseaList$p<0.05)*((MouseSFRCgseaList$NES>0)*2-1)
#Mouse NvC
MouseSFNCgsea=MouseGSEADataMerge[,grepl("FNC",colnames(MouseGSEADataMerge))]
MouseSFNCgseaList=list(p=MouseSFNCgsea[,seq(1,ncol(MouseSFNCgsea),3)],NES=MouseSFNCgsea[,seq(2,ncol(MouseSFNCgsea),3)])
MouseSFNCgseaList$Vote=1*(MouseSFNCgseaList$p<0.05)*((MouseSFNCgseaList$NES>0)*2-1)

#Rat
RatGSEADataMerge=readRDS("Input RDS/Rat_Merged_GSEA_df.rds")
rownames(RatGSEADataMerge)=paste(rep("ID",nrow(RatGSEADataMerge)),1:nrow(RatGSEADataMerge),sep ="")
RatGSEADataMergeIDs=dplyr::select(RatGSEADataMerge,pathway)
#Rat TvU All 
RatFTUgsea=RatGSEADataMerge[,grepl("FTU",colnames(RatGSEADataMerge))]
RatFTUgseaList=list(p=RatFTUgsea[,seq(1,ncol(RatFTUgsea),3)],NES=RatFTUgsea[,seq(2,ncol(RatFTUgsea),3)])
RatFTUgseaList$Vote=1*(RatFTUgseaList$p<0.05)*((RatFTUgseaList$NES>0)*2-1)
#Rat TvU No stress
RatXFTUgsea=RatGSEADataMerge[,grepl("XFTU",colnames(RatGSEADataMerge))]
RatXFTUgseaList=list(p=RatXFTUgsea[,seq(1,ncol(RatXFTUgsea),3)],NES=RatXFTUgsea[,seq(2,ncol(RatXFTUgsea),3)])
RatXFTUgseaList$Vote=1*(RatXFTUgseaList$p<0.05)*((RatXFTUgseaList$NES>0)*2-1)
#Rat TvU Stress
RatSFTUgsea=RatGSEADataMerge[,grepl("SFTU",colnames(RatGSEADataMerge))]
RatSFTUgseaList=list(p=RatSFTUgsea[,seq(1,ncol(RatSFTUgsea),3)],NES=RatSFTUgsea[,seq(2,ncol(RatSFTUgsea),3)])
RatSFTUgseaList$Vote=1*(RatSFTUgseaList$p<0.05)*((RatSFTUgseaList$NES>0)*2-1)

#All
AllGSEADataMerge=readRDS("Input RDS/All_Merged_GSEA_df.rds")
rownames(AllGSEADataMerge)=paste(rep("ID",nrow(AllGSEADataMerge)),1:nrow(AllGSEADataMerge),sep ="")
AllGSEADataMergeIDs=dplyr::select(AllGSEADataMerge,pathway)
#All TvU
AllFTUgsea=AllGSEADataMerge[,grepl("FTU",colnames(AllGSEADataMerge))]
AllFTUgseaList=list(p=AllFTUgsea[,seq(1,ncol(AllFTUgsea),3)],NES=AllFTUgsea[,seq(2,ncol(AllFTUgsea),3)])
AllFTUgseaList$Vote=1*(AllFTUgseaList$p<0.05)*((AllFTUgseaList$NES>0)*2-1)
#All TvU No Stress
AllXFTUgsea=AllGSEADataMerge[,grepl("XFTU",colnames(AllGSEADataMerge))]
AllXFTUgseaList=list(p=AllXFTUgsea[,seq(1,ncol(AllXFTUgsea),3)],NES=AllXFTUgsea[,seq(2,ncol(AllXFTUgsea),3)])
AllXFTUgseaList$Vote=1*(AllXFTUgseaList$p<0.05)*((AllXFTUgseaList$NES>0)*2-1)
#All TvU Stress
AllSFTUgsea=AllGSEADataMerge[,grepl("[SD]FTU",colnames(AllGSEADataMerge))]
AllSFTUgseaList=list(p=AllSFTUgsea[,seq(1,ncol(AllSFTUgsea),3)],NES=AllSFTUgsea[,seq(2,ncol(AllSFTUgsea),3)])
AllSFTUgseaList$Vote=1*(AllSFTUgseaList$p<0.05)*((AllSFTUgseaList$NES>0)*2-1)
#All RvN
AllSFRNgsea=AllGSEADataMerge[,grepl("FRN",colnames(AllGSEADataMerge))]
AllSFRNgseaList=list(p=AllSFRNgsea[,seq(1,ncol(AllSFRNgsea),3)],NES=AllSFRNgsea[,seq(2,ncol(AllSFRNgsea),3)])
AllSFRNgseaList$Vote=1*(AllSFRNgseaList$p<0.05)*((AllSFRNgseaList$NES>0)*2-1)
#Brain Only RvN
BrainSFRNgsea=AllGSEADataMerge[,grepl("FRN",colnames(AllGSEADataMerge))&
                              !grepl("GSE84184",colnames(AllGSEADataMerge))&
                              !grepl("GSE83386",colnames(AllGSEADataMerge))]
BrainSFRNgseaList=list(p=BrainSFRNgsea[,seq(1,ncol(BrainSFRNgsea),3)],NES=BrainSFRNgsea[,seq(2,ncol(BrainSFRNgsea),3)])
BrainSFRNgseaList$Vote=1*(BrainSFRNgseaList$p<0.05)*((BrainSFRNgseaList$NES>0)*2-1)
#All RvC
AllSFRCgsea=AllGSEADataMerge[,grepl("FRC",colnames(AllGSEADataMerge))]
AllSFRCgseaList=list(p=AllSFRCgsea[,seq(1,ncol(AllSFRCgsea),3)],NES=AllSFRCgsea[,seq(2,ncol(AllSFRCgsea),3)])
AllSFRCgseaList$Vote=1*(AllSFRCgseaList$p<0.05)*((AllSFRCgseaList$NES>0)*2-1)
#All NvC
AllSFNCgsea=AllGSEADataMerge[,grepl("FNC",colnames(AllGSEADataMerge))]
AllSFNCgseaList=list(p=AllSFNCgsea[,seq(1,ncol(AllSFNCgsea),3)],NES=AllSFNCgsea[,seq(2,ncol(AllSFNCgsea),3)])
AllSFNCgseaList$Vote=1*(AllSFNCgseaList$p<0.05)*((AllSFNCgseaList$NES>0)*2-1)

jaccard <- function(a, b) {
    intersection = length(intersect(a, b))
    union = length(a) + length(b) - intersection
    return (intersection/union)
}

```

## MetaDE Analysis

https://github.com/metaOmics/MetaDE/tree/master/man

Calculate Fisher and Maximum p-value statistics for gene and pathway level groups of comparisons.

## Differential Expression Treated VS Untreated Analysis

```{r Human TvU MetaDE Analysis}
DataList=HumanDFTUDEList

#Method to detect any significant differences
#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Method to detect consistent significant differences
#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/HumanFTUDE.rds")

```

```{r Mouse TvU All MetaDE Analysis}
DataList=MouseFTUDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseFTUDE.rds")

```

```{r Mouse TvU No stress MetaDE Analysis}
DataList=MouseXFTUDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseXFTUDE.rds")

```

```{r Mouse TvU Stress MetaDE Analysis}
DataList=MouseSFTUDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFTUDE.rds")

```

```{r Rat TvU All MetaDE Analysis}
DataList=RatDEDataList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/RatFTUDE.rds")

```

```{r Rat TvU No stress MetaDE Analysis}
DataList=RatXFTUDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/RatXFTUDE.rds")

```

```{r Rat TvU Stress MetaDE Analysis}
DataList=RatSFTUDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/RatSFTUDE.rds")

```

## Differential Expression Responders VS Non-responders 

```{r Mouse RvN MetaDE Analysis}
DataList=MouseSFRNDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFRNDE.rds")

```

```{r Mouse Brain Only RvN MetaDE Analysis}
DataList=BrainMouseSFRNDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/BrainMouseSFRNDE.rds")

```

```{r Mouse RvC MetaDE Analysis}
DataList=MouseSFRCDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFRCDE.rds")

```

```{r Mouse NvC MetaDE Analysis}
DataList=MouseSFNCDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFNCDE.rds")

```

## GSEA Treated VS Untreated Analysis

```{r All TvU All MetaDE Analysis}
DataList=AllFTUgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/AllFTUgsea.rds")

```

```{r All TvU No Stress MetaDE Analysis}
DataList=AllXFTUgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/AllXFTUgsea.rds")

```

```{r All TvU Stress MetaDE Analysis}
DataList=AllSFTUgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/AllSFTUgsea.rds")

```

```{r Human TvU MetaDE Analysis}
DataList=HumanDFTUgseaList
IDref=HumanGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/HumanFTUgsea.rds")

```

```{r Mouse TvU All MetaDE Analysis}
DataList=MouseFTUgseaList
IDref=MouseGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseFTUgsea.rds")

```

```{r Mouse TvU No stress MetaDE Analysis}
DataList=MouseXFTUgseaList
IDref=MouseGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseXFTUgsea.rds")

```

```{r Mouse TvU Stress MetaDE Analysis}
DataList=MouseSFTUgseaList
IDref=MouseGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFTUgsea.rds")

```

```{r Rat TvU All MetaDE Analysis}
DataList=RatFTUgseaList
IDref=RatGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/RatFTUgsea.rds")

```

```{r Rat TvU No stress MetaDE Analysis}
DataList=RatXFTUgseaList
IDref=RatGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/RatXFTUgsea.rds")

```

```{r Rat TvU Stress MetaDE Analysis}
DataList=RatSFTUgseaList
IDref=RatGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/RatSFTUgsea.rds")

```

## GSEA Responders VS Non-responders Analysis

```{r All RvN MetaDE Analysis}
DataList=AllSFRNgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/AllSFRNgsea.rds")

```

```{r Brain Only RvN MetaDE Analysis}
DataList=BrainSFRNgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/BrainSFRNgsea.rds")

```

```{r All RvC MetaDE Analysis}
DataList=AllSFRCgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/AllSFRCgsea.rds")

```

```{r All NvC MetaDE Analysis}
DataList=AllSFNCgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/AllSFNCgsea.rds")

```

```{r Mouse RvN MetaDE Analysis}
DataList=MouseSFRNgseaList
IDref=MouseGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFRNgsea.rds")

```

```{r Mouse RvC MetaDE Analysis}
DataList=MouseSFRCgseaList
IDref=MouseGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFRCgsea.rds")

```

```{r Mouse NvC MetaDE Analysis}
DataList=MouseSFNCgseaList
IDref=MouseGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/MouseSFNCgsea.rds")

```

## Sensitivity Analyses

```{r Mouse XFTUDE without GSE150431 Analysis}
#Mouse XFTUDE without GSE150431_Amygdala_XFTUXX (RNA Seq w/ low p-values and all female mice)
SAMouseXFTUDE=MouseXFTUDE[,!grepl("GSE150431",colnames(MouseXFTUDE))]
SAMouseXFTUDEList=list(p=SAMouseXFTUDE[,seq(1,ncol(SAMouseXFTUDE),3)],logFC=SAMouseXFTUDE[,seq(2,ncol(SAMouseXFTUDE),3)])
SAMouseXFTUDEList$Vote=1*(SAMouseXFTUDEList$p<0.05)*((SAMouseXFTUDEList$logFC>0)*2-1)

DataList=SAMouseXFTUDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/SAMouseXFTUDE.rds")

```

```{r Rat All FTU without GSE194289 Analysis}
#Rat TvU w/o brain region study (GSE194289)
SARatXFTUDE=RatDEDataMerge[,!grepl("GSE194289",colnames(RatDEDataMerge))]
SARatXFTUDEList=list(p=SARatXFTUDE[,seq(2,ncol(SARatXFTUDE),3)],logFC=SARatXFTUDE[,seq(3,ncol(SARatXFTUDE),3)])
SARatXFTUDEList$Vote=1*(SARatXFTUDEList$p<0.05)*((SARatXFTUDEList$logFC>0)*2-1)

DataList=SARatXFTUDEList

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Gene=rownames(pvaluesDF)
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/SARatXFTUDE.rds")

```

```{r All No stress without GSE194289 Analysis}
#All No Stress TvU w/o brain region study (GSE194289)
SAAllXFTUgsea=AllXFTUgsea[,!grepl("GSE194289",colnames(AllXFTUgsea))]
SAAllXFTUgseaList=list(p=SAAllXFTUgsea[,seq(1,ncol(SAAllXFTUgsea),3)],NES=SAAllXFTUgsea[,seq(2,ncol(SAAllXFTUgsea),3)])
SAAllXFTUgseaList$Vote=1*(SAAllXFTUgseaList$p<0.05)*((SAAllXFTUgseaList$NES>0)*2-1)

DataList=SAAllXFTUgseaList
IDref=AllGSEADataMergeIDs

#Fisher
MetaResFish=MetaDE.pvalue(x = DataList,
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryFish=data.frame(ind.p = MetaResFish$ind.p,
                       stat = MetaResFish$meta.analysis$stat[,1], 
                       pval = MetaResFish$meta.analysis$pval[,1], 
                       FDR = MetaResFish$meta.analysis$FDR[,1])

#Maximum P-value
MetaResMax=MetaDE.pvalue(x = DataList,
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
MetaSummaryMax=data.frame(ind.p = MetaResMax$ind.p,
                       stat = MetaResMax$meta.analysis$stat[,1], 
                       pval = MetaResMax$meta.analysis$pval[,1], 
                       FDR = MetaResMax$meta.analysis$FDR[,1])

#Graphs
#Combined p-value table
pvaluesDF=as.data.frame(DataList$p)
pvaluesDF$Freq=rowSums(pvaluesDF<0.05, na.rm = TRUE )
pvaluesDF$Fish=MetaSummaryFish$FDR
pvaluesDF$Max=MetaSummaryMax$FDR
pvaluesDF$Pathway=IDref$pathway[match(rownames(pvaluesDF),rownames(IDref))]
pvaluesDF$Vote=rowSums(DataList$Vote,na.rm = TRUE)
pvaluesDF$Group=ifelse(pvaluesDF$Fish<0.05, ifelse(pvaluesDF$Max<0.05, "Fisher & Maximum p-value","Fisher"),ifelse(pvaluesDF$Max<0.05,"Maximum p-value","Neither"))
pvaluesDF=filter(pvaluesDF,!is.na(Fish) & !is.na(Max))
saveRDS(pvaluesDF,"MetaDE RDS/SAAllXFTUgsea.rds")

```

## MetaDE Results Visualization

Prepare frequency of significant comparisons graphs, vote density plots, and Venn diagrams.

Categories ("Fisher","MaxP","Freq50","Both","Neither")
Colors ("#21B0FE","#FED700","#FE218B","green","grey70")

## Differential Expression Treated VS Untreated Graphs

```{r Human TvU Graphs}
pvaluesDF=readRDS("MetaDE RDS/HumanFTUDE.rds")
DataList=HumanDFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/HumanFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Maximum p-value","Neither"), label=c("Maximum p-value","Neither"),values=c("#FED700","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/HumanFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))
#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("MaxP","Freq50"),
             fill=c(alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#FED700","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#FED700","#FE218B"),
             filename = "Figures/HumanFTUDE_ScaledVenn.tiff")
#Non-scaled Venn
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("MaxP","Freq50"),
             fill=c(alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#FED700","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#FED700","#FE218B"),
             filename = "Figures/HumanFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF,Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/HumanFTUDE_Heatmap.tiff")

```

```{r Mouse TvU All Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseFTUDE.rds")
DataList=MouseFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Fisher & Maximum p-value","Neither"), label=c("Fisher","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF,Max<0.05 | Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseFTUDE_Heatmap.tiff")

```

```{r Mouse TvU No stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseXFTUDE.rds")
DataList=MouseXFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseXFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Neither"), label=c("Fisher","Neither"),values=c("#21B0FE","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseXFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/MouseXFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/MouseXFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseXFTUDE_Heatmap.tiff")

```

```{r Mouse TvU Stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFTUDE.rds")
DataList=MouseSFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=2, 
             cat.cex=2,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFTUDE_Heatmap.tiff")

```

```{r Rat TvU All Graphs}
pvaluesDF=readRDS("MetaDE RDS/RatFTUDE.rds")
DataList=RatDEDataList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/RatFTUDE_Heatmap.tiff")

```

```{r Rat TvU No stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/RatXFTUDE.rds")
DataList=RatXFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatXFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatXFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatXFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatXFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/RatXFTUDE_Heatmap.tiff")

```

```{r Rat TvU Stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/RatSFTUDE.rds")
DataList=RatSFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatSFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Neither"), label=c("Fisher","Neither"),values=c("#21B0FE","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatSFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/RatSFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/RatSFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
#No overlap present

```

## Differential Expression Responders VS Non-responders Graphs

```{r Mouse RvN Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFRNDE.rds")
DataList=MouseSFRNDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRNDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Neither"), label=c("Fisher","Neither"),values=c("#21B0FE","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRNDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/MouseSFRNDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/MouseSFRNDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFRNDE_Heatmap.tiff")

```

```{r Mouse Brain Only RvN Graphs}
pvaluesDF=readRDS("MetaDE RDS/BrainMouseSFRNDE.rds")
DataList=BrainMouseSFRNDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/BrainMouseSFRNDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Neither"), label=c("Fisher","Neither"),values=c("#21B0FE","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/BrainMouseSFRNDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/BrainMouseSFRNDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/BrainMouseSFRNDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/BrainMouseSFRNDE_Heatmap.tiff")

```

```{r Mouse RvC Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFRCDE.rds")
DataList=MouseSFRCDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRCDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRCDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFRCDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFRCDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFRCDE_Heatmap.tiff")

```

```{r Mouse NvC Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFNCDE.rds")
DataList=MouseSFNCDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFNCDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFNCDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFNCDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFNCDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFNCDE_Heatmap.tiff")

```

## GSEA Treated VS Untreated Graphs

```{r All TvU All Graphs}
pvaluesDF=readRDS("MetaDE RDS/AllFTUgsea.rds")
DataList=AllFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),5))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Fisher & Maximum p-value","Neither"), label=c("Fisher","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.2, 0.8))
ggsave("Figures/AllFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Max<0.05 | Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/AllFTUgsea_Heatmap.tiff")

```

```{r All TvU No Stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/AllXFTUgsea.rds")
DataList=AllXFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllXFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllXFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllXFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllXFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Max<0.05 | Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/AllXFTUgsea_Heatmap.tiff")

```

```{r All TvU Stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/AllSFTUgsea.rds")
DataList=AllSFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/AllSFTUgsea_Heatmap.tiff")

```

```{r Human TvU Graphs}
pvaluesDF=readRDS("MetaDE RDS/HumanFTUgsea.rds")
DataList=HumanDFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/HumanFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/HumanFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/HumanFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/HumanFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/HumanFTUgsea_Heatmap.tiff")

```

```{r Mouse TvU All Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseFTUgsea.rds")
DataList=MouseFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Fisher & Maximum p-value","Neither"), label=c("Fisher","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseFTUgsea_Heatmap.tiff")

```

```{r Mouse TvU No stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseXFTUgsea.rds")
DataList=MouseXFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseXFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Fisher & Maximum p-value","Neither"), label=c("Fisher","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseXFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseXFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseXFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseXFTUgsea_Heatmap.tiff")

```

```{r Mouse TvU Stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFTUgsea.rds")
DataList=MouseSFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFTUgsea_Heatmap.tiff")

```

```{r Rat TvU All Graphs}
pvaluesDF=readRDS("MetaDE RDS/RatFTUgsea.rds")
DataList=RatFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/RatFTUgsea_Heatmap.tiff")

```

```{r Rat TvU No stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/RatXFTUgsea.rds")
DataList=RatXFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatXFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatXFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatXFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatXFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/RatXFTUgsea_Heatmap.tiff")

```

```{r Rat TvU Stress Graphs}
pvaluesDF=readRDS("MetaDE RDS/RatSFTUgsea.rds")
DataList=RatSFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatSFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/RatSFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatSFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/RatSFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/RatSFTUgsea_Heatmap.tiff")

```

## GSEA Responders VS Non-responders Graphs

```{r All RvN Graphs}
pvaluesDF=readRDS("MetaDE RDS/AllSFRNgsea.rds")
DataList=AllSFRNgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFRNgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF[pvaluesDF$Group!="Maximum p-value",])+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Fisher & Maximum p-value","Neither"), label=c("Fisher","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFRNgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFRNgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFRNgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/AllSFRNgsea_Heatmap.tiff")

```

```{r Brain Only RvN Graphs}
pvaluesDF=readRDS("MetaDE RDS/BrainSFRNgsea.rds")
DataList=BrainSFRNgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/BrainSFRNgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/BrainSFRNgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/BrainSFRNgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/BrainSFRNgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/BrainSFRNgsea_Heatmap.tiff")

```

```{r All RvC Graphs}
pvaluesDF=readRDS("MetaDE RDS/AllSFRCgsea.rds")
DataList=AllSFRCgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFRCgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFRCgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFRCgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFRCgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/AllSFRCgsea_Heatmap.tiff")

```

```{r All NvC Graphs}
pvaluesDF=readRDS("MetaDE RDS/AllSFNCgsea.rds")
DataList=AllSFNCgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFNCgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/AllSFNCgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFNCgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/AllSFNCgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/AllSFNCgsea_Heatmap.tiff")

```

```{r Mouse RvN Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFRNgsea.rds")
DataList=MouseSFRNgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRNgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRNgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFRNgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFRNgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFRNgsea_Heatmap.tiff")

```

```{r Mouse RvC Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFRCgsea.rds")
DataList=MouseSFRCgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRCgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFRCgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFRCgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFRCgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFRCgsea_Heatmap.tiff")

```

```{r Mouse NvC Graphs}
pvaluesDF=readRDS("MetaDE RDS/MouseSFNCgsea.rds")
DataList=MouseSFNCgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFNCgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"), label=c("Fisher","Maximum p-value","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","#FED700","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/MouseSFNCgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFNCgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/MouseSFNCgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/MouseSFNCgsea_Heatmap.tiff")

```

## Sensitivity Analyses Graphs

```{r Mouse XFTUDE without GSE150431 Graphs}
pvaluesDF=readRDS("MetaDE RDS/SAMouseXFTUDE.rds")
DataList=SAMouseXFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/SAMouseXFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Neither"), label=c("Fisher","Neither"),values=c("#21B0FE","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/SAMouseXFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/SAMouseXFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/SAMouseXFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$logFC)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),7,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/SAMouseXFTUDE_Heatmap.tiff")

```

```{r Mouse XFTUDE without GSE150431 vs Mouse XFTUDE Graphs}
#Mouse XFTUDE without GSE150431_Amygdala_XFTUXX (RNA Seq w/ low p-values)
SApvaluesDF=readRDS("MetaDE RDS/SAMouseXFTUDE.rds")
pvaluesDF=readRDS("MetaDE RDS/MouseXFTUDE.rds")

CombinedPvalues=merge(SApvaluesDF,pvaluesDF,by = "Gene",suffixes = c("_SA","_O"))
rownames(CombinedPvalues)=CombinedPvalues$Gene

#Fisher
ggplot(CombinedPvalues,aes(x=-log10(Fish_O),y=-log10(Fish_SA)))+
  geom_point(color="navy",alpha=0.2)+
  geom_abline(slope = 1, intercept = 0,linewidth=1)+
  theme_classic()+
  xlab("Fisher -log10(FDR) (original)")+
  ylab("Fisher -log10(FDR) (GSE150431 removed)")+
  geom_hline(yintercept = -log10(0.05), linetype = "dashed")+
  geom_vline(xintercept = -log10(0.05), linetype = "dashed")
ggsave("Figures/SAMouseXFTUDE_FisherScatter.tiff",height = 4, width = 5)

ggplot(pivot_longer(CombinedPvalues,cols = c("Fish_O","Fish_SA")),aes(x=-log10(value),fill=name))+
  geom_histogram(position = position_dodge())+
  theme_classic()+
  scale_fill_discrete(name="Fisher -log10(FDR)",label=c("Original","GSE150431 removed"))

#MaxP
ggplot(CombinedPvalues,aes(x=-log10(Max_O),y=-log10(Max_SA)))+
  geom_point(color="navy",alpha=0.2)+
  geom_abline(slope = 1, intercept = 0,linewidth=1)+
  theme_classic()+
  xlab("MaxP -log10(FDR) (original)")+
  ylab("MaxP -log10(FDR) (GSE150431 removed)")+
  geom_hline(yintercept = -log10(0.05), linetype = "dashed")+
  geom_vline(xintercept = -log10(0.05), linetype = "dashed")
ggsave("Figures/SAMouseXFTUDE_MaxPScatter.tiff",height = 4, width = 5)

ggplot(pivot_longer(CombinedPvalues,cols = c("Max_O","Max_SA")),aes(x=-log10(value),fill=name))+
  geom_histogram(binwidth = 0.1,position = position_dodge())+
  theme_classic()+
  scale_fill_discrete(name="MaxP -log10(FDR)",label=c("Original","GSE150431 removed"))

FisherVennList=list(Fisher_SA=rownames(CombinedPvalues[CombinedPvalues$Fish_SA<0.05,]),
                    Fisher_O=rownames(CombinedPvalues[CombinedPvalues$Fish_O<0.05,]))
                 
#Scaled Venn
venn.diagram(FisherVennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher_SA","Fisher_O"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/SAMouseXFTUDE_FisherScaledVenn.tiff")

```

```{r Rat All FTU without GSE194289 Graphs}
pvaluesDF=readRDS("MetaDE RDS/SARatXFTUDE.rds")
DataList=SARatXFTUDEList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Genes")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/SARatXFTUDE_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Neither"), label=c("Fisher","Neither"),values=c("#21B0FE","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/SARatXFTUDE_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/SARatXFTUDE_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/SARatXFTUDE_Venn.tiff")

#Heatmap for genes in Venn overlap
#No genes in the overlap

```

```{r Rat All FTU without GSE194289 vs Rat All FTU Graphs}
#Rat TvU w/o brain region study (GSE194289)
SApvaluesDF=readRDS("MetaDE RDS/SARatXFTUDE.rds")
pvaluesDF=readRDS("MetaDE RDS/RatFTUDE.rds")

CombinedPvalues=merge(SApvaluesDF,pvaluesDF,by = "Gene",suffixes = c("_SA","_O"))
rownames(CombinedPvalues)=CombinedPvalues$Gene

#Fisher
ggplot(CombinedPvalues,aes(x=-log10(Fish_O),y=-log10(Fish_SA)))+
  geom_point(color="navy",alpha=0.2)+
  geom_abline(slope = 1, intercept = 0,linewidth=1)+
  theme_classic()+
  xlab("Fisher -log10(FDR) (original)")+
  ylab("Fisher -log10(FDR) (GSE194289 removed)")+
  geom_hline(yintercept = -log10(0.05), linetype = "dashed")+
  geom_vline(xintercept = -log10(0.05), linetype = "dashed")
ggsave("Figures/SARatXFTUDE_FisherScatter.tiff",height = 4, width = 5)

ggplot(pivot_longer(CombinedPvalues,cols = c("Fish_O","Fish_SA")),aes(x=-log10(value),fill=name))+
  geom_histogram(position = position_dodge())+
  theme_classic()+
  scale_fill_discrete(name="Fisher -log10(FDR)",label=c("Original","GSE194289 removed"))

#MaxP
ggplot(CombinedPvalues,aes(x=-log10(Max_O),y=-log10(Max_SA)))+
  geom_point(color="navy",alpha=0.2)+
  geom_abline(slope = 1, intercept = 0,linewidth=1)+
  theme_classic()+
  xlab("MaxP -log10(FDR) (original)")+
  ylab("MaxP -log10(FDR) (GSE194289 removed)")+
  geom_hline(yintercept = -log10(0.05), linetype = "dashed")+
  geom_vline(xintercept = -log10(0.05), linetype = "dashed")
ggsave("Figures/SARatXFTUDE_MaxPScatter.tiff",height = 4, width = 5)

ggplot(pivot_longer(CombinedPvalues,cols = c("Max_O","Max_SA")),aes(x=-log10(value),fill=name))+
  geom_histogram(binwidth = 0.1,position = position_dodge())+
  theme_classic()+
  scale_fill_discrete(name="MaxP -log10(FDR)",label=c("Original","GSE194289 removed"))

FisherVennList=list(Fisher_SA=rownames(CombinedPvalues[CombinedPvalues$Fish_SA<0.05,]),
                    Fisher_O=rownames(CombinedPvalues[CombinedPvalues$Fish_O<0.05,]))
                 
#Scaled Venn
venn.diagram(FisherVennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher_SA","Fisher_O"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE, 
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/SARatXFTUDE_FisherScaledVenn.tiff")

```

```{r All No stress without GSE194289 Graphs}
pvaluesDF=readRDS("MetaDE RDS/SAAllXFTUgsea.rds")
DataList=SAAllXFTUgseaList

#Bar graph for Fish and MaxP counts for each Freq value
FreqTableSummary=group_by(pvaluesDF,Freq)%>%
  summarise(Overlap=sum(Fish<0.05&Max<0.05),MaxP=sum(Fish>=0.05&Max<0.05),Fisher=sum(Max>=0.05&Fish<0.05))%>%
  dplyr::rename(SignificantComparison=Freq)
ggplot(pivot_longer(data = FreqTableSummary,cols = c(Overlap,MaxP,Fisher)),aes(x=as.factor(SignificantComparison),y=value,fill=name))+
  geom_col(position = position_dodge(),color="black",linewidth=0.8)+
  theme_classic()+
  scale_x_discrete(limits=as.character(seq(0,nrow(FreqTableSummary),1)),breaks=as.character(seq(0,nrow(FreqTableSummary),1)))+
  scale_fill_manual(name="Significant in:",label=c("Fisher","Maximum p-value","Fisher & Maximum p-value"),values=c("#21B0FE","#FED700","green"))+
  ylab("Pathways")+
  xlab("Number of Significant Comparisons")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/SAAllXFTUgsea_Bar.tiff",height = 5,width = 6)

#Density graph for Fish and MaxP counts for each Vote value
ggplot(pvaluesDF)+
  geom_density(aes(x=Vote,fill=Group),alpha=0.3,linewidth=0.8)+
  theme_classic()+
  scale_x_continuous(breaks=seq(min(pvaluesDF$Vote),max(pvaluesDF$Vote),1))+ #breaks=seq(-X,X,X)
  scale_y_continuous(expand = c(0, 0), limits = c(0, NA))+
  scale_fill_manual(name="Significant in:",limits= c("Fisher","Fisher & Maximum p-value","Neither"), label=c("Fisher","Fisher & Maximum p-value","Neither"),values=c("#21B0FE","green","grey70"))+
  xlab("Expression Change Agreement (Vote)")+
  ylab("Density")+
  theme(axis.text = element_text(size=12),axis.title = element_text(size = 12),legend.position = c(0.8, 0.8))
ggsave("Figures/SAAllXFTUgsea_Density.tiff",height = 5,width = 6)

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>(length(DataList[[1]])/2),])))

#Venn Scaled
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/SAAllXFTUgsea_ScaledVenn.tiff")
#Venn Non-scaled
venn.diagram(VennList,
             euler.d = FALSE, scaled=FALSE, 
             category=c("Fisher","MaxP","Freq50"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B"),
             filename = "Figures/SAAllXFTUgsea_Venn.tiff")

#Heatmap for genes in Venn overlap
logvaluesDF=as.data.frame(DataList$NES)
colnames(logvaluesDF)=ReplaceLabels$ShortLabel[match(str_sub(colnames(logvaluesDF),5,-1),ReplaceLabels$LongLabel)]
pheatmap(logvaluesDF[match(rownames(filter(pvaluesDF, Fish<0.05 & Max<0.05 & Freq>(length(DataList[[1]])/2))),rownames(logvaluesDF)),],
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         filename = "Figures/SAAllXFTUgsea_Heatmap.tiff")

```

```{r All No stress without GSE194289 vs All No stress Graphs}
#Rat TvU w/o brain region study (GSE194289)
SApvaluesDF=readRDS("MetaDE RDS/SAAllXFTUgsea.rds")
pvaluesDF=readRDS("MetaDE RDS/AllXFTUgsea.rds")

CombinedPvalues=merge(SApvaluesDF,pvaluesDF,by = "Pathway",suffixes = c("_SA","_O"))
rownames(CombinedPvalues)=CombinedPvalues$Gene

#Fisher
ggplot(CombinedPvalues,aes(x=-log10(Fish_O),y=-log10(Fish_SA)))+
  geom_point(color="navy",alpha=0.2)+
  geom_abline(slope = 1, intercept = 0,linewidth=1)+
  theme_classic()+
  xlab("Fisher -log10(FDR) (original)")+
  ylab("Fisher -log10(FDR) (GSE194289 removed)")+
  geom_hline(yintercept = -log10(0.05), linetype = "dashed")+
  geom_vline(xintercept = -log10(0.05), linetype = "dashed")
ggsave("Figures/SAAllXFTUgsea_FisherScatter.tiff",height = 4, width = 5)

ggplot(pivot_longer(CombinedPvalues,cols = c("Fish_O","Fish_SA")),aes(x=-log10(value),fill=name))+
  geom_histogram(position = position_dodge())+
  theme_classic()+
  scale_fill_discrete(name="Fisher -log10(FDR)",label=c("Original","GSE194289 removed"))

#MaxP
ggplot(CombinedPvalues,aes(x=-log10(Max_O),y=-log10(Max_SA)))+
  geom_point(color="navy",alpha=0.2)+
  geom_abline(slope = 1, intercept = 0,linewidth=1)+
  theme_classic()+
  xlab("MaxP -log10(FDR) (original)")+
  ylab("MaxP -log10(FDR) (GSE194289 removed)")+
  geom_hline(yintercept = -log10(0.05), linetype = "dashed")+
  geom_vline(xintercept = -log10(0.05), linetype = "dashed")
ggsave("Figures/SAAllXFTUgsea_MaxPScatter.tiff",height = 4, width = 5)

ggplot(pivot_longer(CombinedPvalues,cols = c("Max_O","Max_SA")),aes(x=-log10(value),fill=name))+
  geom_histogram(binwidth = 0.1,position = position_dodge())+
  theme_classic()+
  scale_fill_discrete(name="MaxP -log10(FDR)",label=c("Original","GSE194289 removed"))

FisherVennList=list(Fisher_SA=rownames(CombinedPvalues[CombinedPvalues$Fish_SA<0.05,]),
                    Fisher_O=rownames(CombinedPvalues[CombinedPvalues$Fish_O<0.05,]))
                 
#Scaled Venn
venn.diagram(FisherVennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher_SA","Fisher_O"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4)), 
             col=c("#21B0FE","#FE218B"),
             lty=rep("solid", 2), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,2),
             margin=0.1,
             disable.logging=TRUE, 
             cat.col=c("#21B0FE","#FE218B"),
             filename = "Figures/SAAllXFTUgsea_FisherScaledVenn.tiff")

```

## GSEA Heatmap

Annotated heatmaps for significant pathways.

```{r Load and Prepare Annotations}
npgColors=readRDS("npgColors.rds")
#https://nanx.me/ggsci/reference/pal_npg.html
#show_col(npgColors)
names(npgColors)=c("Signal Transduction","Metabolism of proteins","Immune System","Cell Cycle","Metabolism of RNA",
                   "Neuronal System","KEGG Database","Gene expression (Transcription)","Metabolism","Other") 
StressColors=npgColors[c(2,5,10)]
names(StressColors)=c("Stressed","Unstressed","Depressed")
OrganismColors=npgColors[c(3,4,8)]
names(OrganismColors)=c("Human","Mouse","Rat")
TissueColors=npgColors[c(1,6,7,9)]
names(TissueColors)=c("Cortical","Hippocampus","Blood","Other")

AnnColors=list(Stress=StressColors,Organism=OrganismColors,Tissue=TissueColors,PathwayCategory=npgColors)

ReactomeTop=readRDS("Reactome_HS_withTopLevel.rds")

AllGSEADataMergeIDs$Path=gsub(" \\|.*","",AllGSEADataMergeIDs$pathway)
AllGSEADataMergeIDs$DB=str_match(AllGSEADataMergeIDs$pathway," \\| (.*)")[,2]
AllGSEADataMergeIDs$Top=ReactomeTop$top_level_name[match(gsub(" \\| Reactome","",AllGSEADataMergeIDs$pathway),ReactomeTop$Name)]
AllGSEADataMergeIDs=mutate(AllGSEADataMergeIDs,DBorTop=coalesce(Top,DB))

TissueTypes=read.csv("TissueSubtypes.csv",header = TRUE)

```

```{r All RvN Significant pathways heatmap}
#MetaDE output file with Fish & MaxP results
AllSFRNdf=readRDS("MetaDE RDS/AllSFRNgsea.rds")
#Input file with NES values
AllSFRNNES=AllSFRNgseaList$NES
#Filter NES dataframe to only include significant pathways
AllSFRNNES=AllSFRNNES[match(rownames(AllSFRNdf[AllSFRNdf$Fish<0.05,]),rownames(AllSFRNNES)),]
colnames(AllSFRNNES)=str_sub(colnames(AllSFRNNES),5,-1)
#Annotations
ColumnAnnotations=data.frame(Stress=str_sub(colnames(AllSFRNNES),-6,-6),
                            Organism=c(rep("Mouse",8),"Human"),
                            Tissue=TissueTypes$Subtype[match(str_match(colnames(AllSFRNNES),"_(.*)_")[,2],
                                                             TissueTypes$Abbreviation)])%>%
  mutate(Stress=recode(Stress,"S"="Stressed","X"="Unstressed","D"="Depressed"))%>%
  mutate(Tissue=replace(Tissue,!(Tissue %in% names(TissueColors[1:3])),"Other"))
colnames(AllSFRNNES)=ReplaceLabels$ShortLabel[match(colnames(AllSFRNNES),ReplaceLabels$LongLabel)]
rownames(ColumnAnnotations) = colnames(AllSFRNNES)

#MainCats=c("Immune System","Metabolism of proteins","Gene expression (Transcription)","KEGG","Signal Transduction","Metabolism")
RowAnnotations=data.frame(PathwayCategory=AllGSEADataMergeIDs$DBorTop[match(rownames(AllSFRNNES),rownames(AllGSEADataMergeIDs))])%>%
  mutate(PathwayCategory=replace(PathwayCategory,PathwayCategory=="KEGG","KEGG Database"))%>% 
  mutate(PathwayCategory=replace(PathwayCategory,!(PathwayCategory %in% names(npgColors[1:9])),"Other"))
rownames(RowAnnotations) = rownames(AllSFRNNES)

#Heatmap
pheatmap(AllSFRNNES,
         show_rownames = FALSE,
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         annotation_col = ColumnAnnotations,
         annotation_row = RowAnnotations,
         annotation_colors = AnnColors,
         width = 6,
         filename = "Figures/AllSFRN_SigPaths_Heatmap.tiff")

#Correlations
AllSFRNNEScor=cor(AllSFRNNES%>%na.omit)
pheatmap(AllSFRNNEScor,
         show_rownames = TRUE,
         scale = "none",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         breaks = seq(-1,1,0.04),
         annotation_col = ColumnAnnotations,
         annotation_colors = AnnColors,
         treeheight_row = 0,
         filename = "Figures/AllSFRN_Corr_Heatmap.tiff")

#Jaccard matrix
OnlypvaluesDF=AllSFRNdf[,grepl("Value",colnames(AllSFRNdf))]
SigList=list()
for (i in 1:ncol(OnlypvaluesDF)) {
  SigList[[i]]=rownames(OnlypvaluesDF[na.omit(OnlypvaluesDF[,i])<0.05,])
}
names(SigList)=gsub("P.Value_","",colnames(OnlypvaluesDF))
Compares=expand.grid(names(SigList),names(SigList))
JaccardValues=c()
for (i in 1:nrow(Compares)) {
  JaccardValues=c(JaccardValues,jaccard(SigList[[Compares[i,1]]],SigList[[Compares[i,2]]]))
}
Compares$jaccard=JaccardValues
CompareMatrix=as.data.frame(pivot_wider(Compares,names_from = Var2,values_from = jaccard))
rownames(CompareMatrix)=CompareMatrix$Var1
CompareMatrix=CompareMatrix[,-1]
colnames(CompareMatrix)=ReplaceLabels$ShortLabel[match(colnames(CompareMatrix),ReplaceLabels$LongLabel)]
rownames(CompareMatrix)=ReplaceLabels$ShortLabel[match(rownames(CompareMatrix),ReplaceLabels$LongLabel)]

#Jaccard Heatmap
pheatmap(CompareMatrix,
         show_rownames = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","red"))(20),
         breaks = seq(0,1,0.05),
         annotation_col = ColumnAnnotations,
         annotation_colors = AnnColors,
         treeheight_row = 0,
         filename = "Figures/AllSFRN_Jaccard_Heatmap.tiff")

```

```{r All RvN Max-P Significant pathways heatmap}
#MetaDE output file with Fish & MaxP results
AllSFRNdf=readRDS("MetaDE RDS/AllSFRNgsea.rds")
#Input file with NES values
AllSFRNNES=AllSFRNgseaList$NES
#Filter NES dataframe to only include significant pathways
mAllSFRNNES=AllSFRNNES[match(rownames(AllSFRNdf[AllSFRNdf$Max<0.05,]),rownames(AllSFRNNES)),]
colnames(mAllSFRNNES)=str_sub(colnames(mAllSFRNNES),5,-1)
#Annotations
ColumnAnnotations=data.frame(Stress=str_sub(colnames(mAllSFRNNES),-6,-6),
                            Organism=c(rep("Mouse",8),"Human"),
                            Tissue=TissueTypes$Subtype[match(str_match(colnames(mAllSFRNNES),"_(.*)_")[,2],
                                                             TissueTypes$Abbreviation)])%>%
  mutate(Stress=recode(Stress,"S"="Stressed","X"="Unstressed","D"="Depressed"))%>%
  mutate(Tissue=replace(Tissue,!(Tissue %in% names(TissueColors[1:3])),"Other"))
colnames(mAllSFRNNES)=ReplaceLabels$ShortLabel[match(colnames(mAllSFRNNES),ReplaceLabels$LongLabel)]
rownames(ColumnAnnotations) = colnames(mAllSFRNNES)

RowAnnotations=data.frame(PathwayCategory=AllGSEADataMergeIDs$DBorTop[match(rownames(mAllSFRNNES),rownames(AllGSEADataMergeIDs))])%>%
  mutate(PathwayCategory=replace(PathwayCategory,PathwayCategory=="KEGG","KEGG Database"))%>% 
  mutate(PathwayCategory=replace(PathwayCategory,!(PathwayCategory %in% names(npgColors[1:9])),"Other"))
rownames(RowAnnotations) = rownames(mAllSFRNNES)
rownames(RowAnnotations) = AllGSEADataMergeIDs$Path[match(rownames(RowAnnotations),rownames(AllGSEADataMergeIDs))]
rownames(mAllSFRNNES) = rownames(RowAnnotations)

#Heatmap
pheatmap(mAllSFRNNES,
         show_rownames = TRUE,
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         annotation_col = ColumnAnnotations,
         annotation_row = RowAnnotations,
         annotation_colors = AnnColors,
         width = 10.5,
         filename = "Figures/MaxpAllSFRN_SigPaths_Heatmap.tiff")

```

```{r All TvU Significant pathways heatmap}
#MetaDE output file with Fish & MaxP results
AllFTUdf=readRDS("MetaDE RDS/AllFTUgsea.rds")
#Input file with NES values
AllFTUNES=AllFTUgseaList$NES
#Filter NES dataframe to only include significant pathways
AllFTUNES=AllFTUNES[match(rownames(AllFTUdf[AllFTUdf$Fish<0.05,]),rownames(AllFTUNES)),]
colnames(AllFTUNES)=str_sub(colnames(AllFTUNES),5,-1)
#Annotations
ColumnAnnotations=data.frame(Stress=str_sub(colnames(AllFTUNES),-6,-6),
                            Organism=c(rep("Mouse",20),rep("Rat",33),rep("Human",2)),
                            Tissue=TissueTypes$Subtype[match(str_match(colnames(AllFTUNES),"_(.*)_")[,2],
                                                             TissueTypes$Abbreviation)])%>%
  mutate(Stress=recode(Stress,"S"="Stressed","X"="Unstressed","D"="Depressed"))%>%
  mutate(Tissue=replace(Tissue,!(Tissue %in% names(TissueColors[1:3])),"Other"))
colnames(AllFTUNES)=ReplaceLabels$ShortLabel[match(colnames(AllFTUNES),ReplaceLabels$LongLabel)]
rownames(ColumnAnnotations) = colnames(AllFTUNES)

RowAnnotations=data.frame(PathwayCategory=AllGSEADataMergeIDs$DBorTop[match(rownames(AllFTUNES),rownames(AllGSEADataMergeIDs))])%>%
  mutate(PathwayCategory=replace(PathwayCategory,PathwayCategory=="KEGG","KEGG Database"))%>% 
  mutate(PathwayCategory=replace(PathwayCategory,!(PathwayCategory %in% names(npgColors[1:9])),"Other"))
rownames(RowAnnotations) = rownames(AllFTUNES)

#Heatmap
pheatmap(AllFTUNES,
         show_rownames = FALSE,
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         annotation_col = ColumnAnnotations,
         annotation_row = RowAnnotations,
         annotation_colors = AnnColors,
         width = 12,
         filename = "Figures/AllFTU_SigPaths_Heatmap.tiff")

#Correlations
AllFTUNEScor=cor(AllFTUNES%>%na.omit)
pheatmap(AllFTUNEScor,
         show_rownames = TRUE,
         scale = "none",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         breaks = seq(-1,1,0.04),
         annotation_col = ColumnAnnotations,
         annotation_colors = AnnColors,
         treeheight_row = 0,
         filename = "Figures/AllFTU_Corr_Heatmap.tiff")

#Jaccard matrix
OnlypvaluesDF=AllFTUdf[,grepl("Value",colnames(AllFTUdf))]
SigList=list()
for (i in 1:ncol(OnlypvaluesDF)) {
  SigList[[i]]=rownames(OnlypvaluesDF[na.omit(OnlypvaluesDF[,i])<0.05,])
}
names(SigList)=gsub("P.Value_","",colnames(OnlypvaluesDF))
Compares=expand.grid(names(SigList),names(SigList))
JaccardValues=c()
for (i in 1:nrow(Compares)) {
  JaccardValues=c(JaccardValues,jaccard(SigList[[Compares[i,1]]],SigList[[Compares[i,2]]]))
}
Compares$jaccard=JaccardValues
CompareMatrix=as.data.frame(pivot_wider(Compares,names_from = Var2,values_from = jaccard))
rownames(CompareMatrix)=CompareMatrix$Var1
CompareMatrix=CompareMatrix[,-1]
colnames(CompareMatrix)=ReplaceLabels$ShortLabel[match(colnames(CompareMatrix),ReplaceLabels$LongLabel)]
rownames(CompareMatrix)=ReplaceLabels$ShortLabel[match(rownames(CompareMatrix),ReplaceLabels$LongLabel)]

#Jaccard Heatmap
pheatmap(CompareMatrix,
         show_rownames = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","red"))(20),
         breaks = seq(0,1,0.05),
         annotation_col = ColumnAnnotations,
         annotation_colors = AnnColors,
         treeheight_row = 0,
         filename = "Figures/AllFTU_Jaccard_Heatmap.tiff")

```

```{r All TvU Max-P Significant pathways heatmap}
#MetaDE output file with Fish & MaxP results
AllFTUdf=readRDS("MetaDE RDS/AllFTUgsea.rds")
#Input file with NES values
AllFTUNES=AllFTUgseaList$NES
#Filter NES dataframe to only include significant pathways
mAllFTUNES=AllFTUNES[match(rownames(AllFTUdf[AllFTUdf$Max<0.05,]),rownames(AllFTUNES)),]
colnames(mAllFTUNES)=str_sub(colnames(mAllFTUNES),5,-1)
#Annotations
ColumnAnnotations=data.frame(Stress=str_sub(colnames(mAllFTUNES),-6,-6),
                            Organism=c(rep("Mouse",20),rep("Rat",33),rep("Human",2)),
                            Tissue=TissueTypes$Subtype[match(str_match(colnames(mAllFTUNES),"_(.*)_")[,2],
                                                             TissueTypes$Abbreviation)])%>%
  mutate(Stress=recode(Stress,"S"="Stressed","X"="Unstressed","D"="Depressed"))%>%
  mutate(Tissue=replace(Tissue,!(Tissue %in% names(TissueColors[1:3])),"Other"))
colnames(mAllFTUNES)=ReplaceLabels$ShortLabel[match(colnames(mAllFTUNES),ReplaceLabels$LongLabel)]
rownames(ColumnAnnotations) = colnames(mAllFTUNES)

RowAnnotations=data.frame(PathwayCategory=AllGSEADataMergeIDs$DBorTop[match(rownames(mAllFTUNES),rownames(AllGSEADataMergeIDs))])%>%
  mutate(PathwayCategory=replace(PathwayCategory,PathwayCategory=="KEGG","KEGG Database"))%>%
  mutate(PathwayCategory=replace(PathwayCategory,PathwayCategory=="Reactome","Signal Transduction"))%>%
  mutate(PathwayCategory=replace(PathwayCategory,PathwayCategory=="Muscle contraction","Other"))
rownames(RowAnnotations) = rownames(mAllFTUNES)
rownames(RowAnnotations) = AllGSEADataMergeIDs$Path[match(rownames(RowAnnotations),rownames(AllGSEADataMergeIDs))]
rownames(mAllFTUNES) = rownames(RowAnnotations)

#Heatmap
pheatmap(mAllFTUNES,
         show_rownames = TRUE,
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         annotation_col = ColumnAnnotations,
         annotation_row = RowAnnotations,
         annotation_colors = AnnColors,
         width = 15,
         filename = "Figures/MaxpAllFTU_SigPaths_Heatmap.tiff")

```

```{r Stress TvU Fisher Significant pathways heatmap}
#MetaDE output file with Fish & MaxP results
AllSFTUdf=readRDS("MetaDE RDS/AllSFTUgsea.rds")
#Input file with NES values
AllSFTUNES=AllSFTUgseaList$NES
#Filter NES dataframe to only include significant pathways
AllSFTUNES=AllSFTUNES[match(rownames(AllSFTUdf[AllSFTUdf$Fish<0.05,]),rownames(AllSFTUNES)),]
colnames(AllSFTUNES)=str_sub(colnames(AllSFTUNES),5,-1)
#Annotations
ColumnAnnotations=data.frame(Stress=str_sub(colnames(AllSFTUNES),-6,-6),
                            Organism=c(rep("Mouse",7),rep("Rat",4),rep("Human",2)),
                            Tissue=TissueTypes$Subtype[match(str_match(colnames(AllSFTUNES),"_(.*)_")[,2],
                                                             TissueTypes$Abbreviation)])%>%
  mutate(Stress=recode(Stress,"S"="Stressed","X"="Unstressed","D"="Depressed"))%>%
  mutate(Tissue=replace(Tissue,!(Tissue %in% names(TissueColors[1:3])),"Other"))
colnames(AllSFTUNES)=ReplaceLabels$ShortLabel[match(colnames(AllSFTUNES),ReplaceLabels$LongLabel)]
rownames(ColumnAnnotations) = colnames(AllSFTUNES)

RowAnnotations=data.frame(PathwayCategory=AllGSEADataMergeIDs$DBorTop[match(rownames(AllSFTUNES),rownames(AllGSEADataMergeIDs))])%>%
  mutate(PathwayCategory=replace(PathwayCategory,PathwayCategory=="KEGG","KEGG Database"))%>% 
  mutate(PathwayCategory=replace(PathwayCategory,!(PathwayCategory %in% names(npgColors[1:9])),"Other"))
rownames(RowAnnotations) = rownames(AllSFTUNES)

#Heatmap
pheatmap(AllSFTUNES,
         show_rownames = FALSE,
         scale = "row",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         annotation_col = ColumnAnnotations,
         annotation_row = RowAnnotations,
         annotation_colors = AnnColors,
         width = 7,
         filename = "Figures/AllSFTU_SigPaths_Heatmap.tiff")

#Correlations
AllSFTUNEScor=cor(AllSFTUNES%>%na.omit)
pheatmap(AllSFTUNEScor,
         show_rownames = TRUE,
         scale = "none",
         color = colorRampPalette(c("navy", "white", "red"))(50),
         breaks = seq(-1,1,0.04),
         annotation_col = ColumnAnnotations,
         annotation_colors = AnnColors,
         treeheight_row = 0,
         filename = "Figures/AllSFTU_Corr_Heatmap.tiff")

#Jaccard matrix
OnlypvaluesDF=AllSFTUdf[,grepl("Value",colnames(AllSFTUdf))]
SigList=list()
for (i in 1:ncol(OnlypvaluesDF)) {
  SigList[[i]]=rownames(OnlypvaluesDF[na.omit(OnlypvaluesDF[,i])<0.05,])
}
names(SigList)=gsub("P.Value_","",colnames(OnlypvaluesDF))
Compares=expand.grid(names(SigList),names(SigList))
JaccardValues=c()
for (i in 1:nrow(Compares)) {
  JaccardValues=c(JaccardValues,jaccard(SigList[[Compares[i,1]]],SigList[[Compares[i,2]]]))
}
Compares$jaccard=JaccardValues
CompareMatrix=as.data.frame(pivot_wider(Compares,names_from = Var2,values_from = jaccard))
rownames(CompareMatrix)=CompareMatrix$Var1
CompareMatrix=CompareMatrix[,-1]
colnames(CompareMatrix)=ReplaceLabels$ShortLabel[match(colnames(CompareMatrix),ReplaceLabels$LongLabel)]
rownames(CompareMatrix)=ReplaceLabels$ShortLabel[match(rownames(CompareMatrix),ReplaceLabels$LongLabel)]

#Jaccard Heatmap
pheatmap(CompareMatrix,
         show_rownames = TRUE,
         scale = "none",
         color = colorRampPalette(c("white","red"))(20),
         breaks = seq(0,1,0.05),
         annotation_col = ColumnAnnotations,
         annotation_colors = AnnColors,
         treeheight_row = 0,
         filename = "Figures/AllSFTU_Jaccard_Heatmap.tiff")

```

## Gene Set Overlap

Overlap between differentially regulated genes and previously generated gene set lists of depression related genes.

```{r Load Gene Sets}
ensembl=useEnsembl(biomart="genes",dataset = "mmusculus_gene_ensembl")
#attributes = listAttributes(ensembl)
MouseGeneList=getBM(mart = ensembl,attributes = c("hsapiens_homolog_associated_gene_name","external_gene_name"))%>%filter(hsapiens_homolog_associated_gene_name!="")

DepressionGeneHuman=read.csv("Wu2021_MDD_Metaanalysis.csv",header = TRUE)%>%dplyr::select(GeneSymbol=gene_symbol)
DepressionGeneMouse=data.frame(GeneSymbol=MouseGeneList$external_gene_name[which(MouseGeneList$hsapiens_homolog_associated_gene_name %in% DepressionGeneHuman$GeneSymbol)])

GeneSightHuman=read.csv("GeneSightList.csv",header = TRUE)%>%dplyr::select(GeneSymbol)
GeneSightMouse=data.frame(GeneSymbol=MouseGeneList$external_gene_name[which(MouseGeneList$hsapiens_homolog_associated_gene_name %in% GeneSightHuman$GeneSymbol)])

```

```{r MouseFTUDE}
pvaluesDF=readRDS("MetaDE RDS/MouseFTUDE.rds")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              DepressionGenes=DepressionGeneMouse$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50","DepressionGenes"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             lty=rep("solid", 4), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,4),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             filename = "Figures/MouseFTUDE_DepressionGenes_ScaledVenn.tiff")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              GeneSight=GeneSightMouse$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50","GeneSight"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             lty=rep("solid", 4), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,4),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             filename = "Figures/MouseFTUDE_GeneSight_ScaledVenn.tiff")

```

```{r MouseSFTUDE}
pvaluesDF=readRDS("MetaDE RDS/MouseSFTUDE.rds")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              DepressionGenes=DepressionGeneMouse$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50","DepressionGenes"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             lty=rep("solid", 4), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,4),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             filename = "Figures/MouseSFTUDE_DepressionGenes_ScaledVenn.tiff")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              GeneSight=GeneSightMouse$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","MaxP","Freq50","GeneSight"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FED700",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             lty=rep("solid", 4), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,4),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FED700","#FE218B","#05f505"),
             filename = "Figures/MouseSFTUDE_GeneSight_ScaledVenn.tiff")

```

```{r MouseSFRNDE}
pvaluesDF=readRDS("MetaDE RDS/MouseSFRNDE.rds")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              DepressionGenes=DepressionGeneMouse$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50","DepressionGenes"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#21B0FE","#FE218B","#05f505"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE, 
             cat.col=c("#21B0FE","#FE218B","#05f505"),
             filename = "Figures/MouseSFRNDE_DepressionGenes_ScaledVenn.tiff")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              GeneSight=GeneSightMouse$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Fisher","Freq50","GeneSight"),
             fill=c(alpha("#21B0FE",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#21B0FE","#FE218B","#05f505"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#21B0FE","#FE218B","#05f505"),
             filename = "Figures/MouseSFRNDE_GeneSight_ScaledVenn.tiff")

```

```{r HumanFTUDE}
pvaluesDF=readRDS("MetaDE RDS/HumanFTUDE.rds")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              DepressionGenes=DepressionGeneHuman$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("MaxP","Freq50","DepressionGenes"),
             fill=c(alpha("#FED700",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#FED700","#FE218B","#05f505"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#FED700","#FE218B","#05f505"),
             filename = "Figures/HumanFTUDE_DepressionGenes_ScaledVenn.tiff")

#Venn diagram for Fish, MaxP, and Freq50 (significant in at least 50% of comparisons)
VennList=list_drop_empty(list(Fisher=rownames(pvaluesDF[pvaluesDF$Fish<0.05,]),
                              MaxP=rownames(pvaluesDF[pvaluesDF$Max<0.05,]),
                              Frequency50=rownames(pvaluesDF[pvaluesDF$Freq>((ncol(pvaluesDF)-6)/2),]),
                              GeneSight=GeneSightHuman$GeneSymbol))

#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("MaxP","Freq50","GeneSight"),
             fill=c(alpha("#FED700",0.4),alpha("#FE218B",0.4),alpha("#05f505",0.4)), 
             col=c("#FED700","#FE218B","#05f505"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#FED700","#FE218B","#05f505"),
             filename = "Figures/HumanFTUDE_GeneSight_ScaledVenn.tiff")

```

## Significant Genes/Pathways Bar Graph and Scatter Plot

Bar graphs displaying the number of significant genes and pathways for each comparison.

```{r Load and Combine Organism Data for All Graphs}
TissueTypes=read.csv("TissueSubtypes.csv",header = TRUE)
npgColors=readRDS("npgColors.rds")
StressOrganismColors=npgColors[c(3,4,2,8,5)]
names(StressOrganismColors)=c("MDD Human","Stressed Mouse","Unstressed Mouse","Stressed Rat","Unstressed Rat")
TissueShapes=c(1,2,3,0)
names(TissueShapes)=c("Cortical","Hippocampus","Blood","Other")

##DIFF EXPRESSION
#Load Data
diff_exp_hs <- readRDS("Input RDS/Human_All_DE_List.rds")
diff_exp_mm <- readRDS("Input RDS/Mouse_All_DE_List.rds")
diff_exp_rn <- readRDS("Input RDS/Rat_All_DE_List.rds")
#Process Data
diff_up_hs=data.frame(comparison=names(diff_exp_hs),Freq=sapply(diff_exp_hs,function(x){sum(x[,4]<0.05&x[,3]>0)}))
diff_down_hs=data.frame(comparison=names(diff_exp_hs),Freq=sapply(diff_exp_hs,function(x){sum(x[,4]<0.05&x[,3]<0)}))
diff_up_mm=data.frame(comparison=names(diff_exp_mm),Freq=sapply(diff_exp_mm,function(x){sum(x[,4]<0.05&x[,3]>0)}))
diff_down_mm=data.frame(comparison=names(diff_exp_mm),Freq=sapply(diff_exp_mm,function(x){sum(x[,4]<0.05&x[,3]<0)}))
diff_up_rn=data.frame(comparison=names(diff_exp_rn),Freq=sapply(diff_exp_rn,function(x){sum(x[,4]<0.05&x[,3]>0)}))
diff_down_rn=data.frame(comparison=names(diff_exp_rn),Freq=sapply(diff_exp_rn,function(x){sum(x[,4]<0.05&x[,3]<0)}))

diff_Combined_up_counts=rbind(diff_up_hs,diff_up_mm,diff_up_rn)
diff_Combined_down_counts=rbind(diff_down_hs,diff_down_mm,diff_down_rn)
diff_Combined_merge=cbind(diff_Combined_up_counts,diff_Combined_down_counts)
diff_Combined_merge$counts=rowSums(diff_Combined_merge[,c(2,4)])
diff_Combined_merge$samples=ReplaceLabels$SamplesPerGroup[match(gsub("_DE_df","",diff_Combined_merge$comparison),ReplaceLabels$LongLabel)]
diff_Combined_merge=diff_Combined_merge[,c(1,5,6)]
diff_Combined_merge$Stress=str_sub(diff_Combined_merge$comparison,-12,-12)
diff_Combined_merge=mutate(diff_Combined_merge,Stress=recode(Stress,"S"="Stressed","X"="Unstressed","D"="MDD"))
diff_Combined_merge$Organism=c(rep("Human",5),rep("Mouse",44),rep("Rat",33))
diff_Combined_merge$StressOrganism=paste(diff_Combined_merge$Stress,diff_Combined_merge$Organism)
diff_Combined_merge$TorR=str_sub(diff_Combined_merge$comparison,-10,-9)
diff_Combined_merge$Tissue=TissueTypes$Subtype[match(str_match(rownames(diff_Combined_merge),"_(.*)_.*_DE_df")[,2],TissueTypes$Abbreviation)]
diff_Combined_merge=mutate(diff_Combined_merge,Tissue=replace(Tissue,!(Tissue %in% names(TissueShapes)),"Other"))
diff_Combined_merge$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_Combined_merge),1,-7),ReplaceLabels$LongLabel)]

##PATHWAYS (GSEA)
#Load Data
GSEAhs <- readRDS("Input RDS/Human_All_GSEA_List.rds")
GSEAmm <- readRDS("Input RDS/Mouse_All_GSEA_List.rds")
GSEArn <- readRDS("Input RDS/Rat_All_GSEA_List.rds")
#Process Data
GSEA_up_hs=data.frame(comparison=names(GSEAhs),Freq=sapply(GSEAhs,function(x){sum(x[,4]<0.05&x[,3]>0,na.rm = TRUE)}))
GSEA_down_hs=data.frame(comparison=names(GSEAhs),Freq=sapply(GSEAhs,function(x){sum(x[,4]<0.05&x[,3]<0,na.rm = TRUE)}))
GSEA_up_mm=data.frame(comparison=names(GSEAmm),Freq=sapply(GSEAmm,function(x){sum(x[,4]<0.05&x[,3]>0,na.rm = TRUE)}))
GSEA_down_mm=data.frame(comparison=names(GSEAmm),Freq=sapply(GSEAmm,function(x){sum(x[,4]<0.05&x[,3]<0,na.rm = TRUE)}))
GSEA_up_rn=data.frame(comparison=names(GSEArn),Freq=sapply(GSEArn,function(x){sum(x[,4]<0.05&x[,3]>0,na.rm = TRUE)}))
GSEA_down_rn=data.frame(comparison=names(GSEArn),Freq=sapply(GSEArn,function(x){sum(x[,4]<0.05&x[,3]<0,na.rm = TRUE)}))

GSEA_Combined_up_counts=rbind(GSEA_up_hs,GSEA_up_mm,GSEA_up_rn)
GSEA_Combined_down_counts=rbind(GSEA_down_hs,GSEA_down_mm,GSEA_down_rn)
GSEA_Combined_merge=cbind(GSEA_Combined_up_counts,GSEA_Combined_down_counts)
GSEA_Combined_merge$counts=rowSums(GSEA_Combined_merge[,c(2,4)])
GSEA_Combined_merge$samples=ReplaceLabels$SamplesPerGroup[match(gsub("_GSEA","",GSEA_Combined_merge$comparison),ReplaceLabels$LongLabel)]
GSEA_Combined_merge=GSEA_Combined_merge[,c(1,5,6)]
GSEA_Combined_merge$Stress=str_sub(GSEA_Combined_merge$comparison,-11,-11)
GSEA_Combined_merge=mutate(GSEA_Combined_merge,Stress=recode(Stress,"S"="Stressed","X"="Unstressed","D"="MDD"))
GSEA_Combined_merge$Organism=c(rep("Human",5),rep("Mouse",44),rep("Rat",33))
GSEA_Combined_merge$StressOrganism=paste(GSEA_Combined_merge$Stress,GSEA_Combined_merge$Organism)
GSEA_Combined_merge$TorR=str_sub(GSEA_Combined_merge$comparison,-9,-8)
GSEA_Combined_merge$Tissue=TissueTypes$Subtype[match(str_match(rownames(GSEA_Combined_merge),"_(.*)_.*_GSEA")[,2],TissueTypes$Abbreviation)]
GSEA_Combined_merge=mutate(GSEA_Combined_merge,Tissue=replace(Tissue,!(Tissue %in% names(TissueShapes)),"Other"))
GSEA_Combined_merge$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_Combined_merge),1,-6),ReplaceLabels$LongLabel)]

```

```{r Scatter Plot}
#Differential Expression
ggplot(diff_Combined_merge,aes(samples,counts,color=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  #scale_y_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Genes")+
  theme_classic()+
  geom_text_repel(max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffGenesScatter.png",height = 5,width = 6)

#Response
#No scale
ggplot(diff_Combined_merge[diff_Combined_merge$TorR=="RN",],aes(samples,counts,color=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Genes")+
  theme_classic()+
  geom_text_repel(max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffGenesResponseScatter.png",height = 5,width = 6)

#Log scale
ggplot(diff_Combined_merge[diff_Combined_merge$TorR=="RN",],aes(samples,counts,color=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  scale_y_continuous(transform = "log10")+
  scale_x_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Genes")+
  theme_classic()+
  geom_text_repel(max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffGenesResponseScatterLog.png",height = 5,width = 6)

#Treatment
#No scale
ggplot(diff_Combined_merge[diff_Combined_merge$TorR=="TU",],aes(samples,counts,color=StressOrganism,fill=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Genes")+
  theme_classic()+
  geom_text_repel(max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffGenesTreatmentScatter.png",height = 5,width = 6)

#Log scale
ggplot(diff_Combined_merge[diff_Combined_merge$TorR=="TU",],aes(samples,counts,color=StressOrganism,fill=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  scale_y_continuous(transform = "log10")+
  scale_x_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Genes")+
  theme_classic()+
  geom_text_repel(data = filter(diff_Combined_merge[diff_Combined_merge$TorR=="TU",],counts>6),max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffGenesTreatmentScatterLog.png",height = 5,width = 6)

#GSEA
ggplot(GSEA_Combined_merge,aes(samples,counts,color=StressOrganism,fill=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  #scale_y_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Pathways")+  
  theme_classic()+
  geom_text_repel(max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffPathsScatter.png",height = 5,width = 6)

#Response
#No scale
ggplot(GSEA_Combined_merge[GSEA_Combined_merge$TorR=="RN",],aes(samples,counts,color=StressOrganism,fill=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  #scale_y_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+
  xlab("Samples per Group")+
  ylab("Number of Significant Pathways")+  
  theme_classic()+
  geom_text_repel(max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffPathsResponseScatter.png",height = 5,width = 6)

#Log scale
ggplot(GSEA_Combined_merge[GSEA_Combined_merge$TorR=="RN",],aes(samples,counts,color=StressOrganism,fill=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  scale_y_continuous(transform = "log10")+
  scale_x_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+
  xlab("Samples per Group")+
  ylab("Number of Significant Pathways")+  
  theme_classic()+
  geom_text_repel(data = filter(GSEA_Combined_merge[GSEA_Combined_merge$TorR=="RN",],counts>6),max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffPathsResponseScatterLog.png",height = 5,width = 6)

#Treatment
#No scale
ggplot(GSEA_Combined_merge[GSEA_Combined_merge$TorR=="TU",],aes(samples,counts,color=StressOrganism,fill=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  #scale_y_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Pathways")+  
  theme_classic()+
  geom_text_repel(max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffPathsTreatmentScatter.png",height = 5,width = 6)

#Log scale
ggplot(GSEA_Combined_merge[GSEA_Combined_merge$TorR=="TU",],aes(samples,counts,color=StressOrganism,fill=StressOrganism,shape=factor(Tissue,levels =c("Cortical","Hippocampus","Blood","Other")),label=comparison))+
  geom_point(size=5,stroke=2)+
  scale_y_continuous(transform = "log10")+
  scale_x_continuous(transform = "log10")+
  scale_color_manual(values = StressOrganismColors)+
  scale_shape_manual(name="Tissue",values = c(1,2,3,0))+  
  xlab("Samples per Group")+
  ylab("Number of Significant Pathways")+  
  theme_classic()+
  geom_text_repel(data = filter(GSEA_Combined_merge[GSEA_Combined_merge$TorR=="TU",],counts>6),max.overlaps = 20,size=3,box.padding   = 0.5, point.padding = 0.5)
ggsave("Figures/DiffPathsTreatmentScatterLog.png",height = 5,width = 6)

```

```{r Load and Prepare Annotations}
npgColors=readRDS("npgColors.rds")
#https://nanx.me/ggsci/reference/pal_npg.html
#show_col(npgColors)
names(npgColors)=c("Signal Transduction","Metabolism of proteins","Immune System","Cell Cycle","Metabolism of RNA",
                   "Neuronal System","KEGG Database","Gene expression (Transcription)","Metabolism","Other") 
StressColors=npgColors[c(2,5,10)]
names(StressColors)=c("Stressed","Unstressed","Depressed")
OrganismColors=npgColors[c(3,4,8)]
names(OrganismColors)=c("Human","Mouse","Rat")
TissueColors=npgColors[c(1,6,7,9)]
names(TissueColors)=c("Cortical","Hippocampus","Blood","Other")

AnnColors=list(Stress=StressColors,Organism=OrganismColors,Tissue=TissueColors,PathwayCategory=npgColors)

ReactomeTop=readRDS("Reactome_HS_withTopLevel.rds")

AllGSEADataMergeIDs$Path=gsub(" \\|.*","",AllGSEADataMergeIDs$pathway)
AllGSEADataMergeIDs$DB=str_match(AllGSEADataMergeIDs$pathway," \\| (.*)")[,2]
AllGSEADataMergeIDs$Top=ReactomeTop$top_level_name[match(gsub(" \\| Reactome","",AllGSEADataMergeIDs$pathway),ReactomeTop$Name)]
AllGSEADataMergeIDs=mutate(AllGSEADataMergeIDs,DBorTop=coalesce(Top,DB))

TissueTypes=read.csv("TissueSubtypes.csv",header = TRUE)

```

```{r Human}
##DIFF EXPRESSION
#Load Data
diff_exp <- readRDS("Input RDS/Human_All_DE_List.rds")
#Process Data
diff_exp_p_grp_up_counts=data.frame(comparison=names(diff_exp),Freq=sapply(diff_exp,function(x){sum(x[,4]<0.05&x[,3]>0)}))
diff_exp_p_grp_down_counts=data.frame(comparison=names(diff_exp),Freq=sapply(diff_exp,function(x){sum(x[,4]<0.05&x[,3]<0)}))

#Make graph components
#No significant differentially regulated genes for the human studies

##PATHWAYS (GSEA)
#Load Data
GSEA <- readRDS("Input RDS/Human_All_GSEA_List.rds")
#Process Data
GSEA_p_grp_up_counts=data.frame(comparison=names(GSEA),Freq=sapply(GSEA,function(x){sum(x[,4]<0.05&x[,3]>0,na.rm = TRUE)}))
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts=data.frame(comparison=names(GSEA),Freq=sapply(GSEA,function(x){sum(x[,4]<0.05&x[,3]<0,na.rm = TRUE)}))
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = comparison, y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(250,0),breaks = seq(200,0,-50))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,550),breaks = seq(0,500,100))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/HumanPathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)))


```

```{r Mouse}
##DIFF EXPRESSION
#Load Data
diff_exp <- readRDS("Input RDS/Mouse_All_DE_List.rds")
#Process Data
diff_exp_p_grp_up_counts=data.frame(comparison=names(diff_exp),Freq=sapply(diff_exp,function(x){sum(x[,4]<0.05&x[,3]>0)}))
diff_exp_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_up_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_down_counts=data.frame(comparison=names(diff_exp),Freq=sapply(diff_exp,function(x){sum(x[,4]<0.05&x[,3]<0)}))
diff_exp_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_down_counts),1,-7),ReplaceLabels$LongLabel)]

#Make graph components
g.mid<-ggplot(diff_exp_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1 <- ggplot(data = diff_exp_p_grp_down_counts, aes(x = comparison, y = Freq))+
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Genes Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse()+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2 <- ggplot(data = diff_exp_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Genes Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,4800),breaks = seq(0,4000,1000))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))
grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/MouseGenesSig.png",arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9)),height = 9)

##PATHWAYS (GSEA)
#Load Data
GSEA <- readRDS("Input RDS/Mouse_All_GSEA_List.rds")
#Process Data
GSEA_p_grp_up_counts=data.frame(comparison=names(GSEA),Freq=sapply(GSEA,function(x){sum(x[,4]<0.05&x[,3]>0,na.rm = TRUE)}))
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts=data.frame(comparison=names(GSEA),Freq=sapply(GSEA,function(x){sum(x[,4]<0.05&x[,3]<0,na.rm = TRUE)}))
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = comparison, y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(320,0),breaks = seq(300,0,-100))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,550),breaks = seq(0,500,100))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/MousePathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)),height = 9)

```

```{r Rat}
##DIFF EXPRESSION
#Load Data
diff_exp <- readRDS("Input RDS/Rat_All_DE_List.rds")
#Process Data
diff_exp_p_grp_up_counts=data.frame(comparison=names(diff_exp),Freq=sapply(diff_exp,function(x){sum(x[,4]<0.05&x[,3]>0)}))
diff_exp_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_up_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_down_counts=data.frame(comparison=names(diff_exp),Freq=sapply(diff_exp,function(x){sum(x[,4]<0.05&x[,3]<0)}))
diff_exp_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_down_counts),1,-7),ReplaceLabels$LongLabel)]

#Make graph components
g.mid<-ggplot(diff_exp_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1 <- ggplot(data = diff_exp_p_grp_down_counts, aes(x = comparison, y = Freq))+
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Genes Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse()+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2 <- ggplot(data = diff_exp_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Genes Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,150),breaks = seq(0,125,25))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))
grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/RatGenesSig.png",arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9)),height = 9)

##PATHWAYS (GSEA)
#Load Data
GSEA <- readRDS("Input RDS/Rat_All_GSEA_List.rds")
#Process Data
GSEA_p_grp_up_counts=data.frame(comparison=names(GSEA),Freq=sapply(GSEA,function(x){sum(x[,4]<0.05&x[,3]>0,na.rm = TRUE)}))
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts=data.frame(comparison=names(GSEA),Freq=sapply(GSEA,function(x){sum(x[,4]<0.05&x[,3]<0,na.rm = TRUE)}))
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = comparison, y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(140,0),breaks = seq(125,0,-25))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,140),breaks = seq(0,125,25))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/RatPathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)),height = 9)

```

```{r All Comparisons}
##DIFF EXPRESSION
#Load Data
diff_exp_p_grp_up_counts=diff_Combined_up_counts
diff_exp_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_up_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_down_counts=diff_Combined_down_counts
diff_exp_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_down_counts),1,-7),ReplaceLabels$LongLabel)]

#Make graph components
g.mid<-ggplot(diff_exp_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1 <- ggplot(data = diff_exp_p_grp_down_counts, aes(x = comparison, y = Freq))+
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Genes Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse()+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2 <- ggplot(data = diff_exp_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Genes Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,4800),breaks = seq(0,4000,1000))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))
grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/AllGenesSig.png",arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9)),height = 11)

##PATHWAYS (GSEA)
#Load Data
GSEA_p_grp_up_counts=GSEA_Combined_up_counts
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts=GSEA_Combined_down_counts
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = comparison, y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(320,0),breaks = seq(300,0,-100))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,550),breaks = seq(0,500,100))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/AllPathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)),height = 11)

```

```{r All RvN}
##DIFF EXPRESSION
#Load Data
diff_exp_p_grp_up_counts=diff_Combined_up_counts[grepl("FRN",diff_Combined_up_counts$comparison),]
diff_exp_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_up_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_up_counts$order=ReplaceLabels$Order[match(diff_exp_p_grp_up_counts$comparison,ReplaceLabels$ShortLabel)]
diff_exp_p_grp_down_counts=diff_Combined_down_counts[grepl("FRN",diff_Combined_down_counts$comparison),]
diff_exp_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_down_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_down_counts$order=ReplaceLabels$Order[match(diff_exp_p_grp_down_counts$comparison,ReplaceLabels$ShortLabel)]

#Make graph components
g.mid<-ggplot(diff_exp_p_grp_up_counts,aes(x=1,y=reorder(comparison,-order)))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1 <- ggplot(data = diff_exp_p_grp_down_counts, aes(x = reorder(comparison,-order), y = Freq))+
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Genes Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse()+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2 <- ggplot(data = diff_exp_p_grp_up_counts, aes(x = reorder(comparison,-order), y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Genes Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  #scale_y_continuous(limits = c(0,4800),breaks = seq(0,4000,1000))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))
grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/RvNGenesSig.png",arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9)))

##PATHWAYS (GSEA)
#Load Data
GSEA_p_grp_up_counts=GSEA_Combined_up_counts[grepl("FRN",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_up_counts$order=ReplaceLabels$Order[match(GSEA_p_grp_up_counts$comparison,ReplaceLabels$ShortLabel)]
GSEA_p_grp_down_counts=GSEA_Combined_down_counts[grepl("FRN",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts$order=ReplaceLabels$Order[match(GSEA_p_grp_down_counts$comparison,ReplaceLabels$ShortLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=reorder(comparison,-order)))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = reorder(comparison,-order), y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(170,0),breaks = seq(150,0,-25))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = reorder(comparison,-order), y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,115),breaks = seq(0,100,25))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/RvNPathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)))

```

```{r All TvU}
##DIFF EXPRESSION
#Load Data
diff_exp_p_grp_up_counts=diff_Combined_up_counts[grepl("FTU",diff_Combined_up_counts$comparison),]
diff_exp_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_up_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_up_counts$order=ReplaceLabels$Order[match(diff_exp_p_grp_up_counts$comparison,ReplaceLabels$ShortLabel)]
diff_exp_p_grp_down_counts=diff_Combined_down_counts[grepl("FTU",diff_Combined_down_counts$comparison),]
diff_exp_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_down_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_down_counts$order=ReplaceLabels$Order[match(diff_exp_p_grp_down_counts$comparison,ReplaceLabels$ShortLabel)]

#Make graph components
g.mid<-ggplot(diff_exp_p_grp_up_counts,aes(x=1,y=reorder(comparison,-order)))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1 <- ggplot(data = diff_exp_p_grp_down_counts, aes(x = reorder(comparison,-order), y = Freq))+
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Genes Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse()+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2 <- ggplot(data = diff_exp_p_grp_up_counts, aes(x = reorder(comparison,-order), y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Genes Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,4800),breaks = seq(0,4000,1000))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))
grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/TvUGenesSig.png",arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9)),height = 10)

##PATHWAYS (GSEA)
#Load Data
GSEA_p_grp_up_counts=GSEA_Combined_up_counts[grepl("FTU",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_up_counts$order=ReplaceLabels$Order[match(GSEA_p_grp_up_counts$comparison,ReplaceLabels$ShortLabel)]
GSEA_p_grp_down_counts=GSEA_Combined_down_counts[grepl("FTU",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts$order=ReplaceLabels$Order[match(GSEA_p_grp_down_counts$comparison,ReplaceLabels$ShortLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=reorder(comparison,-order)))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = reorder(comparison,-order), y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(1100,0),breaks = seq(1000,0,-200))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = reorder(comparison,-order), y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,550),breaks = seq(0,500,100))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/TvUPathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)),height = 10)


```

```{r All TvU Stress}
##DIFF EXPRESSION
#Load Data
diff_exp_p_grp_up_counts=diff_Combined_up_counts[grepl("[SD]FTU",diff_Combined_up_counts$comparison),]
diff_exp_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_up_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_down_counts=diff_Combined_down_counts[grepl("[SD]FTU",diff_Combined_down_counts$comparison),]
diff_exp_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_down_counts),1,-7),ReplaceLabels$LongLabel)]

#Make graph components
g.mid<-ggplot(diff_exp_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1 <- ggplot(data = diff_exp_p_grp_down_counts, aes(x = comparison, y = Freq))+
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Genes Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse()+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2 <- ggplot(data = diff_exp_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Genes Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,3800),breaks = seq(0,3000,1000))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))
grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/stressTvUGenesSig.png",arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9)))

##PATHWAYS (GSEA)
#Load Data
GSEA_p_grp_up_counts=GSEA_Combined_up_counts[grepl("[SD]FTU",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts=GSEA_Combined_down_counts[grepl("[SD]FTU",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = comparison, y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(1100,0),breaks = seq(1000,0,-200))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,350),breaks = seq(0,300,100))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/stressTvUPathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)))

```

```{r All TvU No Stress}
##DIFF EXPRESSION
#Load Data
diff_exp_p_grp_up_counts=diff_Combined_up_counts[grepl("XFTU",diff_Combined_up_counts$comparison),]
diff_exp_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_up_counts),1,-7),ReplaceLabels$LongLabel)]
diff_exp_p_grp_down_counts=diff_Combined_down_counts[grepl("XFTU",diff_Combined_down_counts$comparison),]
diff_exp_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(diff_exp_p_grp_down_counts),1,-7),ReplaceLabels$LongLabel)]

#Make graph components
g.mid<-ggplot(diff_exp_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1 <- ggplot(data = diff_exp_p_grp_down_counts, aes(x = comparison, y = Freq))+
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Genes Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse()+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2 <- ggplot(data = diff_exp_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Genes Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,4800),breaks = seq(0,4000,1000))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))
grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/NOstressTvUGenesSig.png",arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(4/9,1/9,4/9)),height = 9)

##PATHWAYS (GSEA)
#Load Data
GSEA_p_grp_up_counts=GSEA_Combined_up_counts[grepl("XFTU",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_up_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_up_counts),1,-6),ReplaceLabels$LongLabel)]
GSEA_p_grp_down_counts=GSEA_Combined_down_counts[grepl("XFTU",GSEA_Combined_up_counts$comparison),]
GSEA_p_grp_down_counts$comparison=ReplaceLabels$ShortLabel[match(str_sub(rownames(GSEA_p_grp_down_counts),1,-6),ReplaceLabels$LongLabel)]

#Make graph components
g.mid_pathway<-ggplot(GSEA_p_grp_up_counts,aes(x=1,y=comparison))+
  geom_text(aes(label=comparison))+
  geom_segment(aes(x=0.94,xend=0.96,yend=comparison))+
  geom_segment(aes(x=1.04,xend=1.065,yend=comparison))+
  ggtitle("")+
  ylab(NULL)+
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065))+
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))

g1_pathway <- ggplot(data = GSEA_p_grp_down_counts, aes(x = comparison, y = Freq)) +
  xlab(NULL)+
  geom_bar(stat = "identity", fill = "#3C5488", color = "black", size = 1 )+
  ggtitle("Pathways Down")+
  theme_classic()+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,-1,1,0), "mm"),
        axis.line.y = element_blank())+
  scale_y_reverse(limits = c(250,0),breaks = seq(200,0,-50))+
  coord_flip()+
  geom_hline(aes(yintercept = 0))

g2_pathway <- ggplot(data = GSEA_p_grp_up_counts, aes(x = comparison, y = Freq))+
  geom_bar(stat = "identity", fill = "#DC0000", color = "black", size = 1 )+
  ggtitle("Pathways Up") +
  theme_classic()+
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size=14),
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5,face = "bold",size = 16),
        plot.margin = unit(c(1,0,1,-1), "mm"), 
        axis.line.y = element_blank()) +
  coord_flip()+
  scale_y_continuous(limits = c(0,550),breaks = seq(0,500,100))+
  geom_hline(aes(yintercept = 0))

#Combine Graphs
gg1_pathway <- ggplot_gtable(ggplot_build(g1_pathway))
gg2_pathway <- ggplot_gtable(ggplot_build(g2_pathway))
gg.mid_pathway <- ggplot_gtable(ggplot_build(g.mid_pathway))
grid.arrange(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9))
ggsave("Figures/NOstressTvUPathsSig.png",arrangeGrob(gg1_pathway,gg.mid_pathway,gg2_pathway,ncol=3,widths=c(4/9,1/9,4/9)),height = 9)


```
