---
title: "Interaction analyses"
author: "Caleb Class"
date: "2024-10-03"
output: html_document
---


Differential expression analyses with interaction terms were conducted in GSE84185Analysis_JK.Rmd

```{r}

# For reproducibility
set.seed(123)

library(Biobase)
source("../AdditionalDatasets/read_cpdb_tab.R")
pathways <- read_cpdb_tab("../FLX RMD/CPDB_pathways_genes_mouse.tab")

lm_fit_Dentate_2 <- readRDS("../FLX_datasets/GSE84183/DE_DG_GSE84183.rds")
lm_fit_Cingulate_2 <- readRDS("../FLX_datasets/GSE84183/DE_CG_GSE84183.rds")
lm_fit_Blood_2 <- readRDS("../FLX_datasets/GSE84184/DE_Blood_GSE84184.rds")

```

Identify genes to use in GSEA

```{r}

library(biomaRt)
#Unique genes
ensembl=useEnsembl(biomart="genes",dataset = "mmusculus_gene_ensembl")
attributes = listAttributes(ensembl)
GeneList=getBM(mart = ensembl,attributes = c("ensembl_gene_id","external_gene_name","ensembl_transcript_id"))

library(dplyr)
gse84_genesymbol <- data.frame(ID=lm_fit_Dentate_2$genes$ID,transcript=lm_fit_Dentate_2$genes$ENSEMBL_ID)
gse84_genesymbol$symbol=GeneList$ensembl_gene_id[match(gse84_genesymbol$transcript,GeneList$ensembl_transcript_id)]
TopTable <- topTable(lm_fit_Dentate_2, coef = 5, number = Inf)
gse84_genesymbol$AvgExpr=TopTable$AveExpr[match(gse84_genesymbol$ID,TopTable$ID)]
gse84_genesymbol=na.omit(gse84_genesymbol)
gse84_genesymbol=arrange(gse84_genesymbol,desc(AvgExpr))%>%filter(!duplicated(symbol))%>%filter(symbol!="")

```


Run GSEA to summarize interaction analysis results by pathway.


```{r}

library(dplyr)

# Dentate
ensembl=useEnsembl(biomart="genes",dataset = "mmusculus_gene_ensembl")
attributes = listAttributes(ensembl)
GeneList=getBM(mart = ensembl,attributes = c("ensembl_gene_id","external_gene_name","ensembl_transcript_id"))

gse84_genesymbol <- data.frame(ID=lm_fit_Dentate_2$genes$ID,transcript=lm_fit_Dentate_2$genes$ENSEMBL_ID)
gse84_genesymbol$symbol=GeneList$ensembl_gene_id[match(gse84_genesymbol$transcript,GeneList$ensembl_transcript_id)]
TopTable <- topTable(lm_fit_Dentate_2, coef = 5, number = Inf)
gse84_genesymbol$AvgExpr=TopTable$AveExpr[match(gse84_genesymbol$ID,TopTable$ID)]
gse84_genesymbol=na.omit(gse84_genesymbol)
gse84_genesymbol=arrange(gse84_genesymbol,desc(AvgExpr))%>%filter(!duplicated(symbol))%>%filter(symbol!="")

DG_Interaction <- TopTable %>% filter(ID %in% gse84_genesymbol$ID)
DG_Interaction$symbol <- gse84_genesymbol$symbol[match(DG_Interaction$ID,gse84_genesymbol$ID)]
DG_Interaction <- DG_Interaction[!is.na(DG_Interaction$symbol),]
DG_Interaction_stat <- DG_Interaction$t
names(DG_Interaction_stat) <- DG_Interaction$symbol
fgseaDG_Interaction <- fgsea::fgsea(pathways, DG_Interaction_stat, minSize=10, nPermSimple=10000)
saveRDS(fgseaDG_Interaction,"../FLX_datasets/GSE84183/GSEA_DG_Interaction_GSE84183.rds")

# Cingulate
gse84_genesymbol <- data.frame(ID=lm_fit_Cingulate_2$genes$ID,transcript=lm_fit_Cingulate_2$genes$ENSEMBL_ID)
gse84_genesymbol$symbol=GeneList$ensembl_gene_id[match(gse84_genesymbol$transcript,GeneList$ensembl_transcript_id)]
TopTable <- topTable(lm_fit_Cingulate_2, coef = 5, number = Inf)
gse84_genesymbol$AvgExpr=TopTable$AveExpr[match(gse84_genesymbol$ID,TopTable$ID)]
gse84_genesymbol=na.omit(gse84_genesymbol)
gse84_genesymbol=arrange(gse84_genesymbol,desc(AvgExpr))%>%filter(!duplicated(symbol))%>%filter(symbol!="")

CG_Interaction <- TopTable %>% filter(ID %in% gse84_genesymbol$ID)
CG_Interaction$symbol <- gse84_genesymbol$symbol[match(CG_Interaction$ID,gse84_genesymbol$ID)]
CG_Interaction <- CG_Interaction[!is.na(CG_Interaction$symbol),]
CG_Interaction_stat <- CG_Interaction$t
names(CG_Interaction_stat) <- CG_Interaction$symbol
fgseaCG_Interaction <- fgsea::fgsea(pathways, CG_Interaction_stat, minSize=10, nPermSimple=10000)
saveRDS(fgseaCG_Interaction,"../FLX_datasets/GSE84183/GSEA_CG_Interaction_GSE84183.rds")

# Blood
gse84_genesymbol <- data.frame(ID=lm_fit_Blood_2$genes$ID,transcript=lm_fit_Blood_2$genes$ENSEMBL_ID)
gse84_genesymbol$symbol=GeneList$ensembl_gene_id[match(gse84_genesymbol$transcript,GeneList$ensembl_transcript_id)]
TopTable <- topTable(lm_fit_Blood_2, coef = 5, number = Inf)
gse84_genesymbol$AvgExpr=TopTable$AveExpr[match(gse84_genesymbol$ID,TopTable$ID)]
gse84_genesymbol=na.omit(gse84_genesymbol)
gse84_genesymbol=arrange(gse84_genesymbol,desc(AvgExpr))%>%filter(!duplicated(symbol))%>%filter(symbol!="")

BL_Interaction <- TopTable %>% filter(ID %in% gse84_genesymbol$ID)
BL_Interaction$symbol <- gse84_genesymbol$symbol[match(BL_Interaction$ID,gse84_genesymbol$ID)]
BL_Interaction <- BL_Interaction[!is.na(BL_Interaction$symbol),]
BL_Interaction_stat <- BL_Interaction$t
names(BL_Interaction_stat) <- BL_Interaction$symbol
fgseaBL_Interaction <- fgsea::fgsea(pathways, BL_Interaction_stat, minSize=10, nPermSimple=10000)
saveRDS(fgseaBL_Interaction,"../FLX_datasets/GSE84184/GSEA_BL_Interaction_GSE84184.rds")


```

Overlap significant gene sets
```{r}

library(VennDiagram)
library(ggvenn)

VennList=list(Cingulate=fgseaCG_Interaction$pathway[fgseaCG_Interaction$padj < 0.05],
                              Dentate=fgseaDG_Interaction$pathway[fgseaDG_Interaction$padj < 0.05],
                              Blood=fgseaBL_Interaction$pathway[fgseaBL_Interaction$padj < 0.05])
#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("CG","DG","Blood"),
             fill=c(alpha("#ffca4e",0.4),alpha("#fc8c78",0.4),alpha("#bf5747",0.4)), 
             col=c("#ffca4e","#fc8c78","#bf5747"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#ffca4e","#fc8c78","#bf5747"),
             filename = "../Figures/Interaction/GSE84185Interaction_ScaledVenn.tiff")
```


```{r}

library(MetaDE)
fgseaBL_Interaction <- readRDS("../FLX_datasets/GSE84184/GSEA_BL_Interaction_GSE84184.rds")
fgseaCG_Interaction <- readRDS("../FLX_datasets/GSE84183/GSEA_CG_Interaction_GSE84183.rds")
fgseaDG_Interaction <- readRDS("../FLX_datasets/GSE84183/GSEA_DG_Interaction_GSE84183.rds")

Interaction_merged <- merge(fgseaBL_Interaction[,c(1:3,6)], fgseaCG_Interaction[,c(1:3,6)], 
                            by = "pathway", suffixes = c(".BL", ".CG"))
Interaction_merged <- merge(Interaction_merged, fgseaDG_Interaction[,c(1:3,6)],
                            by = "pathway", suffixes = c("", ".DG"))

pmat <- Interaction_merged[,c(2,5,8)]
nesmat <- Interaction_merged[,c(4,7,10)]
votemat <- 1*(pmat<0.05)*((nesmat>0)*2-1)
  
MetaResFish <- MetaDE.pvalue(x = list(p=pmat),
                          meta.method = "Fisher",
                          rth = NULL,
                          parametric = TRUE)
Interaction_merged$Fisher <- MetaResFish$meta.analysis$FDR

MetaResMaxP <- MetaDE.pvalue(x = list(p=pmat),
                          meta.method = "maxP",
                          rth = NULL,
                          parametric = TRUE)
Interaction_merged$MaxP <- MetaResMaxP$meta.analysis$FDR

Interaction_merged$Freq <- rowSums(pmat < 0.05)
Interaction_merged$Vote <- rowSums(votemat)


```

Overlap with meta-analysis results within stress & unstress.

```{r}

stressed_meta <- readRDS("../MetaDE RDS/AllSFTUgsea.rds")
unstressed_meta <- readRDS("../MetaDE RDS/AllXFTUgsea.rds")




```


## Differential Expression meta-analysis

```{r}

VennList=list(Interaction=Interaction_merged$pathway[Interaction_merged$MaxP < 0.05 & !is.na(Interaction_merged$MaxP)],
                              Stressed=stressed_meta$Pathway[stressed_meta$Max < 0.05 & !is.na(stressed_meta$Max)],
                              Unstressed=unstressed_meta$Pathway[unstressed_meta$Max < 0.05 & !is.na(unstressed_meta$Max)])
#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Interaction","Stressed","Unstressed"),
             fill=c(alpha("#9f3867",0.4),alpha("#4e6dc8",0.4),alpha("#F39B7F",0.4)), 
             col=c("#9f3867","#4e6dc8","#F39B7F"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.07,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#9f3867","#4e6dc8","#F39B7F"),
             filename = "../Figures/Interaction/MetaAnalysisVsInteraction_ScaledVenn.tiff")


VennList=list(Interaction=Interaction_merged$pathway[Interaction_merged$Fisher < 0.05 & !is.na(Interaction_merged$Fisher)],
                              Stressed=stressed_meta$Pathway[stressed_meta$Fish < 0.05 & !is.na(stressed_meta$Fish)],
                              Unstressed=unstressed_meta$Pathway[unstressed_meta$Fish < 0.05 & !is.na(unstressed_meta$Fish)])
#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Interaction","Stressed","Unstressed"),
             fill=c(alpha("#9f3867",0.4),alpha("#4e6dc8",0.4),alpha("#F39B7F",0.4)), 
             col=c("#9f3867","#4e6dc8","#F39B7F"),
             lty=rep("solid", 3), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.1,3),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#9f3867","#4e6dc8","#F39B7F"),
             filename = "../Figures/Interaction/FisherVsInteraction_ScaledVenn.tiff")

```

```{r}

RN_meta <- readRDS("../MetaDE RDS/AllSFRNgsea.rds")[,10:15]

VennTable <- table(unlist(VennList))
VennTable[VennTable == 3]

mergedResults <- merge(Interaction_merged, stressed_meta[,14:19], 
                       by.x = "pathway", by.y = "Pathway", all = TRUE)
mergedResults <- merge(mergedResults, unstressed_meta[,43:48],
                       by.x = "pathway", by.y = "Pathway",
                       suffixes = c(".stressed", ""), all=TRUE)
mergedResults <- merge(mergedResults, RN_meta,
                       by.x = "pathway", by.y = "Pathway",
                       suffixes = c(".unstressed", ".response"), all=TRUE)

#View(mergedResults)

mergedResults[mergedResults$MaxP < 0.05 & mergedResults$Max.stressed < 0.05]
mergedResults[mergedResults$MaxP < 0.05 & mergedResults$Max.response < 0.05,]
mergedResults[mergedResults$Max.stressed < 0.05 & mergedResults$Max.response < 0.05]


```

Add in R vs. NR

```{r}

VennList=list(Interaction=Interaction_merged$pathway[Interaction_merged$MaxP < 0.05 & !is.na(Interaction_merged$MaxP)],
                              Stressed=stressed_meta$Pathway[stressed_meta$Max < 0.05 & !is.na(stressed_meta$Max)],
                              
              RN = RN_meta$Pathway[RN_meta$Max < 0.05 & !is.na(RN_meta$Pathway)],
              Unstressed=unstressed_meta$Pathway[unstressed_meta$Max < 0.05 & !is.na(unstressed_meta$Max)])
#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Interaction","Stressed","Response", "Unstressed"),
             fill=c(alpha("#9f3867",0.4),alpha("#4e6dc8",0.4),alpha("#00aba1",0.4),alpha("#F39B7F",0.4)), 
             col=c("#9f3867","#4e6dc8", "#00aba1", "#F39B7F"),
             lty=rep("solid", 4), 
             cex=2.3, 
             cat.cex=2.5,
             cat.dist=c(0.3,0.3,0.12,0.12),
             cat.pos=c(210,150,330,30),
             margin=0.2,
             disable.logging=TRUE,
             cat.col=c("#9f3867","#4e6dc8", "#00aba1", "#F39B7F"),
             filename = "../Figures/Interaction/MetaAnalysisRNVsInteraction_ScaledVenn.tiff")


VennList=list(Interaction=Interaction_merged$pathway[Interaction_merged$Fisher < 0.05 & !is.na(Interaction_merged$Fisher)],
                              Stressed=stressed_meta$Pathway[stressed_meta$Fish < 0.05 & !is.na(stressed_meta$Fish)],
              RN = RN_meta$Pathway[RN_meta$Fish < 0.05 & !is.na(RN_meta$Pathway)],
                              Unstressed=unstressed_meta$Pathway[unstressed_meta$Fish < 0.05 & !is.na(unstressed_meta$Fish)]
               )
#Scaled Venn
venn.diagram(VennList,
             euler.d = TRUE, scaled=TRUE, 
             category=c("Interaction","Stressed","Response", "Unstressed"),
             fill=c(alpha("#9f3867",0.4),alpha("#4e6dc8",0.4),alpha("#00aba1",0.4),alpha("#F39B7F",0.4)), 
             col=c("#9f3867","#4e6dc8", "#00aba1", "#F39B7F"),
             lty=rep("solid", 4), 
             cex=3, 
             cat.cex=3,
             cat.dist=rep(0.12,4),
             margin=0.1,
             disable.logging=TRUE,
             cat.col=c("#9f3867","#4e6dc8", "#00aba1", "#F39B7F"),
             filename = "../Figures/Interaction/FisherRNVsInteraction_ScaledVenn.tiff")




```




